---
title: "1 DADA2 Workflow"
description: |
  DADA2 workflow for processing the **16S rRNA** data set. Workflow uses paired end reads, beginning with raw fastq files, ending with sequence & taxonomy tables.
author:
#  - name: Jarrod J Scott
#    url: https://example.com/norajones
#    affiliation: Spacely Sprockets
#    affiliation_url: https://example.com/spacelysprokets
bibliography: assets/cite.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
set.seed(919191)
library(dada2); packageVersion("dada2")
library(ShortRead); packageVersion("ShortRead")
library(Biostrings); packageVersion("Biostrings")
library(ggplot2); packageVersion("ggplot2")
library(DECIPHER); packageVersion("DECIPHER")
library(rmarkdown)
library(DT)
library(gridExtra)
library(grid)

options(scipen=999)

knitr::opts_current$get(c(
  "cache",
  "cache.path",
  "cache.rebuild",
  "dependson",
  "autodep"
))
remove(list = ls())
```

> Hit the *Hide Code* button to collapse the R code (visible by default).

<aside>
```{r codefolder, echo=FALSE, results='asis', eval=TRUE}
codefolder::generic(init = "show", query = "pre.sourceCode",
  style = "position: absolute; right: 14%; z-index: 200")
```
</aside>

# Prerequisites

In order to run this workflow, you either need the *raw* data, available on the figshare project site, or the *trimmed* data, available from the ENA under project accession number ... See the [Data Availability](data-availability.html) page for complete details.

This workflow is how we processed our 16S rRNA data set using [DADA2](https://benjjneb.github.io/dada2/) [@callahan2016dada2]. Workflow construction is based on the [DADA2 Pipeline Tutorial (1.8)](https://benjjneb.github.io/dada2/tutorial_1_8.html) and the primer identification section of the [DADA2 ITS Pipeline Workflow (1.8)](https://benjjneb.github.io/dada2/ITS_workflow.html).

```{r, include=FALSE, eval=TRUE}
remove(list = ls())
```

# Overview

### Sequence Files & Samples

We sequenced a total of ADD DETAILS...

<br/>

```{r, echo=FALSE, layout="l-body-outset", eval=TRUE}
seq_table <- read.table("files/trepo/dada2/tables/ssu_sample_seq_info.txt",
                        header = TRUE, sep = "\t")
datatable(seq_table, width = "100%",
          caption = htmltools::tags$caption(
            style = 'caption-side: bottom; text-align: left;',
            'Table: ', htmltools::em('Sample summary table. Use the buttons to
            navigate through the table or download a copy.')),
          extensions = 'Buttons', options = list(
            scrollX = TRUE,
            dom = 'Blfrtip',
            buttons = c('copy', 'csv', 'excel'),
            pageLength = 5,
            lengthMenu = list(c(5, 10, -1), c("5", "10", "All"))
            )
          )
```

### Workflow

The workflow consists of the following steps:

| Step | What weâ€™re doing                                           | Command                |
|------|------------------------------------------------------------|------------------------|
| 1    | **Remove Primers**                                         |                        |
|      | prepare input file names & paths                           | multiple               |
|      | Define  primers (all orientations)                         | multiple               |
|      | Remove  primers                                            | `cutadapt`             |
| 2    | **Filtering, Error Correction, Dereplication**             |                        |
|      | assess quality & filter reads                              | `filterAndTrim()`      |
|      | generate an error model for the data                       | `learnErrors()`        |
|      | dereplicate sequences                                      | `derepFastq`           |
| 3    | **DADA2 & ASV Inference**                                  |                        |
|      | infer ASVs on both forward & reverse reads independently   | `dada()`               |
| 4    | **Merge Paired Reads & Construct Sequence Tables**         |                        |
|      | merge paired reads                                         | `mergePairs()`         |
|      | generate count tables                                      | `makeSequenceTable()`  |
|      | merge all sequence tables                                  | `mergeSequenceTables()`|
|      | collapse no mismatch                                       | `collapseNoMismatch()` |
| 5    | **Remove Chimeras**                                        |                        |
|      | screen for & remove chimeras                               | `removeBimeraDenovo()` |
| 6    | **Assign Taxonomy**                                        |                        |
|      | assign taxonomy & finish workflow                          | `assignTaxonomy()`     |
| 7    | **Track reads through workflow**                           |                        |


# 1. Remove Primers

## Rename samples

First, we need to setup the working environment by renaming the fastq files and defining a path for the working directory. To make the parsing easier, we will eliminate the lane and well number from each sample. If you do not wish to do that you will need to adjust the code accordingly. 

> CAUTION. You should check that this code works on a few backup files before proceeding.


```{r, warning=FALSE}
setwd("RAW")
orig_fastq <- list.files(pattern = "*.fastq.gz")
newname_fastq <- gsub("_S.*_L001", "", orig_fastq)
newname_fastq <- gsub("_001", "", newname_fastq)
file.rename(orig_fastq, newname_fastq)
setwd("../")
```

```{r}
path <- "RAW"  
head(list.files(path))
```

```{r, echo=FALSE, eval=TRUE}
c("nml001_R1.fastq.gz", "nml001_R2.fastq.gz", "nml002_R1.fastq.gz",
"nml002_R2.fastq.gz", "nml003_R1.fastq.gz", "nml003_R2.fastq.gz")
```

Then, we generate matched lists of the forward and reverse read files. We also parse out the sample name.

```{r}
fnFs <- sort(list.files(path, pattern = "_R1.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern = "_R2.fastq.gz", full.names = TRUE))
```


```{r, layout="l-body-outset", warning=FALSE, fig.height=3}
p1_pre <- plotQualityProfile(fnFs[1:312], aggregate = TRUE)
p2_pre <- plotQualityProfile(fnRs[1:312], aggregate = TRUE)

p3_pre <- grid.arrange(p1_pre, p2_pre, nrow = 1)
ggsave("figures/ssu_raw_plot_qscores.png", p3_pre, width = 7, height = 3)
```


```{r, echo=FALSE, layout="l-body", warning=FALSE, fig.height=4, fig.width=4, eval=TRUE}
system("cp files/trepo/dada2/figures/ssu_raw_plot_qscores.png include/trepo/dada2/ssu_raw_plot_qscores.png")
knitr::include_graphics("include/trepo/dada2/ssu_raw_plot_qscores.png")
```
<small><em>Aggregated quality score plots for raw forward reads & reverse reads.</em></small>

## Define Primers

Before we start the DADA2 workflow we need to run [catadapt](https://github.com/marcelm/cutadapt) [@martin2011cutadapt] on all `fastq.gz` files to trim the primers. For this part of the study we used  the primer pair 505F (GTGCCAGCMGCCGCGGTAA) and 806R   (GGACTACHVGGGTWTCTAAT) [@caporaso2011global] which should yield an amplicon length of about 253 bp.

First we define the primers.

```{r}
FWD <- "GTGYCAGCMGCCGCGGTAA"
REV <- "CCGYCAATTYMTTTRAGTTT"
```

Next, we check the presence and orientation of these primers in the data. I started doing this for ITS data because of primer read-through but I really like the general idea of doing it just to make sure nothing funny is going of with the data.

To do this we will create all orientations of the input primer sequences. In other words the *Forward*, *Complement*, *Reverse*, and *Reverse Complement* variations.

```{r}
allOrients <- function(primer) {
    require(Biostrings)
    dna <- DNAString(primer) 
    orients <- c(Forward = dna, 
                 Complement = complement(dna), 
                 Reverse = reverse(dna), 
                 RevComp = reverseComplement(dna))
    return(sapply(orients, toString))
}
FWD.orients <- allOrients(FWD)
REV.orients <- allOrients(REV)
FWD.orients
REV.orients
```

```
              Forward            Complement               Reverse 
"GTGYCAGCMGCCGCGGTAA" "CACRGTCGKCGGCGCCATT" "AATGGCGCCGMCGACYGTG" 
              RevComp 
"TTACCGCGGCKGCTGRCAC" 
               Forward             Complement                Reverse 
"CCGYCAATTYMTTTRAGTTT" "GGCRGTTAARKAAAYTCAAA" "TTTGARTTTMYTTAACYGCC" 
               RevComp 
"AAACTYAAAKRAATTGRCGG" 
```

```{r}
fnFs.filtN <- file.path(path, basename(fnFs)) 
fnRs.filtN <- file.path(path, basename(fnRs))
```

Nice.  Time to assess the number of times a primer (and all possible primer orientation) appear in the forward and reverse reads. According to the workflow, counting the primers on one set of paired end fastq files is sufficient to see if there is a problem. This assumes that all the files were created using the same library prep. Basically for both primers, we will search for all four orientations in both forward and reverse reads. Since this is 16S rRNA we do not anticipate any issues but it is worth checking.

```{r, echo=FALSE, eval=TRUE}
sampnum <- 2
```


```{r}
primerHits <- function(primer, fn) {
    # Counts number of reads in which the primer is found
    nhits <- vcountPattern(primer, sread(readFastq(fn)), fixed = FALSE)
    return(sum(nhits > 0))
}
```

### Forward primers

```{r}
rbind(FWD.ForwardReads = sapply(FWD.orients, primerHits, fn = fnFs.filtN[[sampnum]]), 
    FWD.ReverseReads = sapply(FWD.orients, primerHits, fn = fnRs.filtN[[sampnum]]))
```

```
                 Forward Complement Reverse RevComp
FWD.ForwardReads   34166          0       0       0
FWD.ReverseReads       0          0       0       1
```

### Reverse primers

```{r}
rbind(REV.ForwardReads = sapply(REV.orients, primerHits, fn = fnFs.filtN[[sampnum]]), 
       REV.ReverseReads = sapply(REV.orients, primerHits, fn = fnRs.filtN[[sampnum]]))
```

```
                 Forward Complement Reverse RevComp
REV.ForwardReads       0          0       0       3
REV.ReverseReads   31194          0       0       0
```

As expected, forward primers predominantly in the forward reads and very little evidence of reverse primers.

## Remove Primers

Now we can run [catadapt](https://github.com/marcelm/cutadapt) [@martin2011cutadapt] to remove the primers from the fastq sequences. A little setup first. If this command executes successfully it means R has found cutadapt.

```{r}
cutadapt <- "/home/scottjj/miniconda3/envs/cutadapt/bin/cutadapt"
system2(cutadapt, args = "--version") # Run shell commands from R
```

```
2.8
```

We set paths and trim the forward primer and the reverse-complement of the reverse primer off of R1 (forward reads) and trim the reverse primer and the reverse-complement of the forward primer off of R2 (reverse reads).

```{r}
path.cut <- file.path(path, "cutadapt")
if(!dir.exists(path.cut)) dir.create(path.cut)
fnFs.cut <- file.path(path.cut, basename(fnFs))
fnRs.cut <- file.path(path.cut, basename(fnRs))

FWD.RC <- dada2:::rc(FWD)
REV.RC <- dada2:::rc(REV)

R1.flags <- paste("-g", FWD, "-a", REV.RC) 
R2.flags <- paste("-G", REV, "-A", FWD.RC) 

for(i in seq_along(fnFs)) {system2(cutadapt, 
                                   args = c(R1.flags, R2.flags, 
                                            "--minimum-length", 250, 
                                            "--times", 2, 
                                            "--error-rate", 0.10, 
                                            "--no-indels",
                                            "--discard-untrimmed",
                                            "--output", fnFs.cut[i], 
                                            "--paired-output", fnRs.cut[i],
                                            "--report minimal",
                                            fnFs.filtN[i], fnRs.filtN[i]))}

```


*Note*. If the code above removes all of the base pairs in a sequence, you will get downstream errors unless you set the `-m` flag. This flag sets the minimum length and reads shorter than this will be discarded. Without this flag, reads of length 0 will be kept and cause issues. *Also, a lot of output will be written to the screen by cutadapt!*.

We can now count the number of primers in the sequences from the output of cutadapt.

```{r}
rbind(
    FWD.ForwardReads = sapply(FWD.orients, primerHits, fn = fnFs.cut[[sampnum]]), 
    FWD.ReverseReads = sapply(FWD.orients, primerHits, fn = fnRs.cut[[sampnum]]), 
    REV.ForwardReads = sapply(REV.orients, primerHits, fn = fnFs.cut[[sampnum]]), 
    REV.ReverseReads = sapply(REV.orients, primerHits, fn = fnRs.cut[[sampnum]]))
```

```
                 Forward Complement Reverse RevComp
FWD.ForwardReads       0          0       0       0
FWD.ReverseReads       0          0       0       0
REV.ForwardReads       0          0       0       0
REV.ReverseReads       0          0       0       0
```

Basically, primers are no longer detected in the cutadapted reads. 

```{r}
p1_cut <- plotQualityProfile(fnFs.cut[1:312], aggregate = TRUE)
p2_cut <- plotQualityProfile(fnRs.cut[1:312], aggregate = TRUE)

p3_cut <- grid.arrange(p1_cut, p2_cut, nrow = 1)
ggsave("figures/ssu_cut_plot_qscores.png", p3_cut, width = 7, height = 3)
```

```{r, echo=FALSE, layout="l-body", warning=FALSE, fig.height=4, fig.width=4, eval=TRUE}
system("cp files/trepo/dada2/figures/ssu_cut_plot_qscores.png include/trepo/dada2/ssu_cut_plot_qscores.png")
knitr::include_graphics("include/trepo/dada2/ssu_cut_plot_qscores.png")
```
<small><em>Aggregated quality score plots for forward reads & reverse reads.</em></small>

Now, for each sample, we can take a look at how primer removal affected the total number of raw reads.

<br/>

```{r echo=FALSE, eval=TRUE}
cut_t <- read.table("files/trepo/dada2/tables/ssu_cut.txt",
                       header = TRUE, row.names = 1)
datatable(cut_t, width = "80%",
          caption = htmltools::tags$caption(
            style = 'caption-side: bottom; text-align: left;',
            'Table: ', htmltools::em('Number of raw reads vs. cutadapt reads')),
          extensions = 'Buttons', options = list(
            scrollX = TRUE,
            dom = 'Blfrtip',
            buttons = c('copy', 'csv', 'excel'),
            pageLength = 5,
            lengthMenu = list(c(5, 10, -1), c("5", "10", "All"))
            )
          )
```


# 2. Filtering, Error Correction, Dereplication

We need the forward and reverse fastq file names and the sample names.

```{r}
cutFs <- sort(list.files(path.cut, pattern = "_R1.fastq.gz", full.names = TRUE))
cutRs <- sort(list.files(path.cut, pattern = "_R2.fastq.gz", full.names = TRUE))

get.sample.name <- function(fname) strsplit(basename(fname), "_")[[1]][1]
sample.names <- unname(sapply(cutFs, get.sample.name))
head(sample.names)
```

```{r, eval=TRUE}
c("nml001", "nml002", "nml003", "nml004", "nml005", "nml006")
```

First let's look at the quality of our reads. The numbers in brackets specify which samples to view. Here we are looking at an aggregate plot of all data (except the negative control)

## Filtering


```{r}
filtFs <- file.path(path.cut, "filtered", basename(cutFs))
filtRs <- file.path(path.cut, "filtered", basename(cutRs))

out <- filterAndTrim(cutFs, filtFs, cutRs, filtRs,  
                     maxN = 0, maxEE = c(3, 5), 
                     truncLen = c(230,170), 
                     truncQ = 2, rm.phix = TRUE, 
                     compress = TRUE, multithread = 20) 
out
```

<aside>
These parameters should be set based on the anticipated length of the amplicon and the read quality.
</aside>

And here is a table of how the filtering step affected the number of reads in each sample. As you can see, there are a few samples that started with a low read count to begin with---we will likely remove those samples at some point.

<br/>

```{r, echo=FALSE, eval=TRUE}
out <- read.table("files/trepo/dada2/tables/ssu_filt.txt",
                       header = TRUE, row.names = 1)
datatable(out, width = "80%",
          caption = htmltools::tags$caption(
            style = 'caption-side: bottom; text-align: left;',
            'Table: ', htmltools::em('Cutadapt reads vs. filtered reads.')),
          extensions = 'Buttons', options = list(
            scrollX = TRUE,
            dom = 'Blfrtip',
            buttons = c('copy', 'csv', 'excel'),
            pageLength = 5,
            lengthMenu = list(c(5, 10, -1), c("5", "10", "All"))
            )
          )
```

```{r, layout="l-body-outset", warning=FALSE, fig.height=3}
p1 <- plotQualityProfile(filtFs[1:312], aggregate = TRUE)
p2 <- plotQualityProfile(filtRs[1:312], aggregate = TRUE)

p3 <- grid.arrange(p1, p2, nrow = 1)
ggsave("figures/ssu_post_filt_plot_qscores.png", p3, width = 7, height = 3)
```


```{r, echo=FALSE, layout="l-body", warning=FALSE, fig.height=4, fig.width=4, eval=TRUE}
system("cp files/trepo/dada2/figures/ssu_post_filt_plot_qscores.png include/trepo/dada2/ssu_post_filt_plot_qscores.png")
knitr::include_graphics("include/trepo/dada2/ssu_post_filt_plot_qscores.png")
```
<small><em>Aggregated quality score plots for filtered forward reads & reverse reads.</em></small>


## Learn Error Rates

Now it is time to assess the error rate of the data. The DADA2 algorithm uses a parametric error model. Every amplicon data set has a different set of error rates and the `learnErrors` method learns this error model *from the data*. It does this by alternating estimation of the error rates and inference of sample composition until they converge on a jointly consistent solution. The algorithm begins with an initial guess, for which the maximum possible error rates in the data are used.

### Forward Reads

```{r}
errF <- learnErrors(filtFs, multithread = TRUE)
```

```
104818820 total bases in 455734 reads from 16 samples will be used for learning the error rates.
```

```{r, warning=FALSE, layout="l-body-outset"}
plotErrors(errF, nominalQ=TRUE)
```

```{r}
p3 <- plotErrors(errF, nominalQ = TRUE)
ggsave("figures/ssu_plot_errorF_1.png", p3, width = 7, height = 5)
ggsave("figures/ssu_plot_errorF_2.png", p3)
```


```{r, echo=FALSE, layout="l-body-outset", eval=TRUE}
system("cp files/trepo/dada2/figures/ssu_plot_errorF_1.png include/trepo/dada2/ssu_plot_errorF_1.png")
knitr::include_graphics("include/trepo/dada2/ssu_plot_errorF_1.png")
```
<small><em>Error plots, forward reads.</em></small>


### Reverse Reads

```{r}
errR <- learnErrors(filtRs, multithread = TRUE)
```

```
102010880 total bases in 600064 reads from 21 samples will be used for learning the error rates.
```

```{r, warning=FALSE, layout="l-body-outset"}
plotErrors(errF, nominalQ=TRUE)
```

```{r}
p4 <- plotErrors(errR, nominalQ = TRUE)
ggsave("figures/ssu_plot_errorR_1.png", p4, width = 7, height = 5)
ggsave("figures/ssu_plot_errorR_2.png", p4)
```


```{r, echo=FALSE, layout="l-body-outset", eval=TRUE}
system("cp files/trepo/dada2/figures/ssu_plot_errorR_1.png include/trepo/dada2/ssu_plot_errorR_1.png")
knitr::include_graphics("include/trepo/dada2/ssu_plot_errorR_1.png")
```
<small><em>Error plots, reverse reads.</em></small>

The error rates for each possible transition (A to C, A to G, etc.) are shown. Points are the observed error rates for each consensus quality score. The black line shows the estimated error rates after convergence of the machine-learning algorithm. The red line shows the error rates expected under the nominal definition of the Q-score. Here the estimated error rates (black line) are a good fit to the observed rates (points), and the error rates drop with increased quality as expected.

## Dereplicate Reads

Now we can use `derepFastq` to identify the unique sequences in the forward and reverse fastq files.

```{r}
derepFs <- derepFastq(filtFs, verbose = TRUE)
derepRs <- derepFastq(filtRs, verbose = TRUE)
```

```{r}
names(derepFs) <- sample.names
names(derepRs) <- sample.names
```

<details markdown="1">
<summary>Detailed results of `derep` forward and reverse reads for</summary>
<pre><code>
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml001_R1.fastq.gz
Encountered 17193 unique sequences from 28541 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml002_R1.fastq.gz
Encountered 17850 unique sequences from 28011 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml003_R1.fastq.gz
Encountered 20482 unique sequences from 32988 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml004_R1.fastq.gz
Encountered 17661 unique sequences from 26821 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml005_R1.fastq.gz
Encountered 19229 unique sequences from 29697 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml006_R1.fastq.gz
Encountered 16925 unique sequences from 24584 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml007_R1.fastq.gz
Encountered 22571 unique sequences from 35704 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml008_R1.fastq.gz
Encountered 15941 unique sequences from 23478 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml009_R1.fastq.gz
Encountered 18508 unique sequences from 29377 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml010_R1.fastq.gz
Encountered 13916 unique sequences from 20444 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml011_R1.fastq.gz
Encountered 10725 unique sequences from 15374 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml012_R1.fastq.gz
Encountered 20125 unique sequences from 30208 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml013_R1.fastq.gz
Encountered 21931 unique sequences from 38160 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml014_R1.fastq.gz
Encountered 19443 unique sequences from 31418 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml015_R1.fastq.gz
Encountered 18851 unique sequences from 31670 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml016_R1.fastq.gz
Encountered 19173 unique sequences from 29259 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml017_R1.fastq.gz
Encountered 20708 unique sequences from 32921 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml018_R1.fastq.gz
Encountered 8896 unique sequences from 12627 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml019_R1.fastq.gz
Encountered 16682 unique sequences from 24680 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml020_R1.fastq.gz
Encountered 18934 unique sequences from 28375 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml021_R1.fastq.gz
Encountered 29363 unique sequences from 45727 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml022_R1.fastq.gz
Encountered 19032 unique sequences from 26948 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml023_R1.fastq.gz
Encountered 24450 unique sequences from 36563 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml024_R1.fastq.gz
Encountered 18963 unique sequences from 27677 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml025_R1.fastq.gz
Encountered 25904 unique sequences from 42203 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml026_R1.fastq.gz
Encountered 21719 unique sequences from 34263 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml027_R1.fastq.gz
Encountered 11782 unique sequences from 17086 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml028_R1.fastq.gz
Encountered 14699 unique sequences from 20692 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml029_R1.fastq.gz
Encountered 13560 unique sequences from 20355 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml030_R1.fastq.gz
Encountered 18514 unique sequences from 28067 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml031_R1.fastq.gz
Encountered 21336 unique sequences from 32440 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml032_R1.fastq.gz
Encountered 15621 unique sequences from 22011 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml033_R1.fastq.gz
Encountered 22780 unique sequences from 35035 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml034_R1.fastq.gz
Encountered 19534 unique sequences from 28303 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml035_R1.fastq.gz
Encountered 24412 unique sequences from 37176 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml036_R1.fastq.gz
Encountered 14666 unique sequences from 21730 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml037_R1.fastq.gz
Encountered 37349 unique sequences from 62880 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml038_R1.fastq.gz
Encountered 23549 unique sequences from 36838 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml039_R1.fastq.gz
Encountered 32398 unique sequences from 53845 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml040_R1.fastq.gz
Encountered 19416 unique sequences from 28445 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml041_R1.fastq.gz
Encountered 17310 unique sequences from 26284 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml042_R1.fastq.gz
Encountered 17057 unique sequences from 25614 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml043_R1.fastq.gz
Encountered 21265 unique sequences from 32664 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml044_R1.fastq.gz
Encountered 23885 unique sequences from 37410 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml045_R1.fastq.gz
Encountered 21188 unique sequences from 32629 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml046_R1.fastq.gz
Encountered 20747 unique sequences from 29876 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml047_R1.fastq.gz
Encountered 19428 unique sequences from 28506 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml048_R1.fastq.gz
Encountered 13609 unique sequences from 18819 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml049_R1.fastq.gz
Encountered 24937 unique sequences from 39117 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml050_R1.fastq.gz
Encountered 24574 unique sequences from 39383 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml051_R1.fastq.gz
Encountered 20126 unique sequences from 31202 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml052_R1.fastq.gz
Encountered 16424 unique sequences from 23204 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml053_R1.fastq.gz
Encountered 27550 unique sequences from 43062 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml054_R1.fastq.gz
Encountered 13869 unique sequences from 18867 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml055_R1.fastq.gz
Encountered 24471 unique sequences from 38385 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml056_R1.fastq.gz
Encountered 22869 unique sequences from 34664 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml057_R1.fastq.gz
Encountered 21597 unique sequences from 31887 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml058_R1.fastq.gz
Encountered 11071 unique sequences from 14906 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml059_R1.fastq.gz
Encountered 9966 unique sequences from 13525 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml060_R1.fastq.gz
Encountered 17519 unique sequences from 26375 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml061_R1.fastq.gz
Encountered 17105 unique sequences from 25137 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml062_R1.fastq.gz
Encountered 17099 unique sequences from 24128 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml063_R1.fastq.gz
Encountered 16225 unique sequences from 23847 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml064_R1.fastq.gz
Encountered 11290 unique sequences from 15571 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml065_R1.fastq.gz
Encountered 11921 unique sequences from 16496 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml066_R1.fastq.gz
Encountered 9231 unique sequences from 12605 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml067_R1.fastq.gz
Encountered 14594 unique sequences from 21047 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml068_R1.fastq.gz
Encountered 10362 unique sequences from 13995 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml069_R1.fastq.gz
Encountered 17298 unique sequences from 24812 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml070_R1.fastq.gz
Encountered 15618 unique sequences from 21059 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml071_R1.fastq.gz
Encountered 12070 unique sequences from 15942 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml072_R1.fastq.gz
Encountered 9946 unique sequences from 12651 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml073_R1.fastq.gz
Encountered 29360 unique sequences from 48494 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml074_R1.fastq.gz
Encountered 11150 unique sequences from 16040 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml075_R1.fastq.gz
Encountered 14078 unique sequences from 20835 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml076_R1.fastq.gz
Encountered 13389 unique sequences from 19162 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml077_R1.fastq.gz
Encountered 23023 unique sequences from 37252 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml078_R1.fastq.gz
Encountered 7245 unique sequences from 9819 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml079_R1.fastq.gz
Encountered 20826 unique sequences from 33832 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml080_R1.fastq.gz
Encountered 8409 unique sequences from 11979 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml081_R1.fastq.gz
Encountered 13982 unique sequences from 20150 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml082_R1.fastq.gz
Encountered 7936 unique sequences from 10707 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml083_R1.fastq.gz
Encountered 17380 unique sequences from 26175 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml084_R1.fastq.gz
Encountered 17689 unique sequences from 25757 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml085_R1.fastq.gz
Encountered 16886 unique sequences from 26498 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml086_R1.fastq.gz
Encountered 17887 unique sequences from 27627 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml087_R1.fastq.gz
Encountered 8087 unique sequences from 11186 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml088_R1.fastq.gz
Encountered 9834 unique sequences from 13337 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml089_R1.fastq.gz
Encountered 50969 unique sequences from 85392 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml090_R1.fastq.gz
Encountered 9120 unique sequences from 12134 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml091_R1.fastq.gz
Encountered 14209 unique sequences from 20730 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml092_R1.fastq.gz
Encountered 15786 unique sequences from 23491 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml093_R1.fastq.gz
Encountered 13674 unique sequences from 19559 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml094_R1.fastq.gz
Encountered 17937 unique sequences from 25711 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml095_R1.fastq.gz
Encountered 14026 unique sequences from 20894 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml096_R1.fastq.gz
Encountered 18415 unique sequences from 25959 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml097_R1.fastq.gz
Encountered 24008 unique sequences from 36632 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml098_R1.fastq.gz
Encountered 30864 unique sequences from 51569 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml099_R1.fastq.gz
Encountered 24304 unique sequences from 38561 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml100_R1.fastq.gz
Encountered 13126 unique sequences from 19340 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml101_R1.fastq.gz
Encountered 13019 unique sequences from 18156 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml102_R1.fastq.gz
Encountered 13669 unique sequences from 19193 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml103_R1.fastq.gz
Encountered 39168 unique sequences from 62024 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml104_R1.fastq.gz
Encountered 21063 unique sequences from 31177 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml105_R1.fastq.gz
Encountered 11560 unique sequences from 16338 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml106_R1.fastq.gz
Encountered 14683 unique sequences from 20974 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml107_R1.fastq.gz
Encountered 8326 unique sequences from 10761 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml108_R1.fastq.gz
Encountered 12568 unique sequences from 17267 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml109_R1.fastq.gz
Encountered 12086 unique sequences from 17517 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml110_R1.fastq.gz
Encountered 10792 unique sequences from 14812 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml111_R1.fastq.gz
Encountered 14724 unique sequences from 20561 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml112_R1.fastq.gz
Encountered 26233 unique sequences from 41817 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml113_R1.fastq.gz
Encountered 10895 unique sequences from 15211 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml114_R1.fastq.gz
Encountered 13098 unique sequences from 18839 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml115_R1.fastq.gz
Encountered 10647 unique sequences from 14752 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml116_R1.fastq.gz
Encountered 14545 unique sequences from 21980 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml117_R1.fastq.gz
Encountered 13079 unique sequences from 18694 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml118_R1.fastq.gz
Encountered 11169 unique sequences from 15305 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml119_R1.fastq.gz
Encountered 15666 unique sequences from 22531 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml120_R1.fastq.gz
Encountered 13085 unique sequences from 19170 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml121_R1.fastq.gz
Encountered 14259 unique sequences from 20736 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml122_R1.fastq.gz
Encountered 12051 unique sequences from 17611 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml123_R1.fastq.gz
Encountered 12196 unique sequences from 17226 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml124_R1.fastq.gz
Encountered 17926 unique sequences from 27681 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml125_R1.fastq.gz
Encountered 3899 unique sequences from 4927 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml126_R1.fastq.gz
Encountered 7955 unique sequences from 10783 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml127_R1.fastq.gz
Encountered 26408 unique sequences from 39001 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml128_R1.fastq.gz
Encountered 19690 unique sequences from 29604 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml129_R1.fastq.gz
Encountered 11142 unique sequences from 15395 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml130_R1.fastq.gz
Encountered 11379 unique sequences from 15946 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml131_R1.fastq.gz
Encountered 11190 unique sequences from 14535 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml132_R1.fastq.gz
Encountered 14990 unique sequences from 20130 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml133_R1.fastq.gz
Encountered 7965 unique sequences from 10486 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml134_R1.fastq.gz
Encountered 43443 unique sequences from 72813 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml135_R1.fastq.gz
Encountered 15026 unique sequences from 21767 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml136_R1.fastq.gz
Encountered 7761 unique sequences from 10371 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml137_R1.fastq.gz
Encountered 16890 unique sequences from 24798 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml138_R1.fastq.gz
Encountered 11762 unique sequences from 16445 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml139_R1.fastq.gz
Encountered 13472 unique sequences from 18935 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml140_R1.fastq.gz
Encountered 16007 unique sequences from 24416 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml141_R1.fastq.gz
Encountered 11855 unique sequences from 16289 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml142_R1.fastq.gz
Encountered 10114 unique sequences from 13046 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml143_R1.fastq.gz
Encountered 10199 unique sequences from 13182 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml144_R1.fastq.gz
Encountered 7598 unique sequences from 9963 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml145_R1.fastq.gz
Encountered 11906 unique sequences from 17346 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml146_R1.fastq.gz
Encountered 14971 unique sequences from 21719 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml147_R1.fastq.gz
Encountered 12014 unique sequences from 17394 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml148_R1.fastq.gz
Encountered 8248 unique sequences from 11488 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml149_R1.fastq.gz
Encountered 11106 unique sequences from 15029 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml150_R1.fastq.gz
Encountered 12343 unique sequences from 17264 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml151_R1.fastq.gz
Encountered 16265 unique sequences from 23194 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml152_R1.fastq.gz
Encountered 26754 unique sequences from 41708 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml153_R1.fastq.gz
Encountered 31821 unique sequences from 49950 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml154_R1.fastq.gz
Encountered 20740 unique sequences from 31188 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml155_R1.fastq.gz
Encountered 22684 unique sequences from 32055 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml156_R1.fastq.gz
Encountered 22631 unique sequences from 32118 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml157_R1.fastq.gz
Encountered 26865 unique sequences from 40929 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml158_R1.fastq.gz
Encountered 24828 unique sequences from 43054 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml159_R1.fastq.gz
Encountered 15718 unique sequences from 22894 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml160_R1.fastq.gz
Encountered 13191 unique sequences from 19298 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml161_R1.fastq.gz
Encountered 15170 unique sequences from 21161 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml162_R1.fastq.gz
Encountered 11782 unique sequences from 16489 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml163_R1.fastq.gz
Encountered 12616 unique sequences from 18268 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml164_R1.fastq.gz
Encountered 14949 unique sequences from 22554 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml165_R1.fastq.gz
Encountered 45 unique sequences from 51 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml166_R1.fastq.gz
Encountered 10107 unique sequences from 13707 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml167_R1.fastq.gz
Encountered 13414 unique sequences from 18048 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml168_R1.fastq.gz
Encountered 4647 unique sequences from 5974 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml169_R1.fastq.gz
Encountered 13176 unique sequences from 19347 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml170_R1.fastq.gz
Encountered 15928 unique sequences from 24081 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml171_R1.fastq.gz
Encountered 56 unique sequences from 63 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml172_R1.fastq.gz
Encountered 11289 unique sequences from 16107 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml173_R1.fastq.gz
Encountered 8220 unique sequences from 10923 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml174_R1.fastq.gz
Encountered 10418 unique sequences from 14282 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml175_R1.fastq.gz
Encountered 8143 unique sequences from 11231 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml176_R1.fastq.gz
Encountered 12384 unique sequences from 17328 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml177_R1.fastq.gz
Encountered 13000 unique sequences from 18237 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml178_R1.fastq.gz
Encountered 11923 unique sequences from 17043 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml179_R1.fastq.gz
Encountered 9327 unique sequences from 12042 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml180_R1.fastq.gz
Encountered 10400 unique sequences from 14477 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml181_R1.fastq.gz
Encountered 7802 unique sequences from 10550 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml182_R1.fastq.gz
Encountered 9448 unique sequences from 13384 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml183_R1.fastq.gz
Encountered 12034 unique sequences from 17422 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml184_R1.fastq.gz
Encountered 11133 unique sequences from 15542 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml185_R1.fastq.gz
Encountered 10672 unique sequences from 14755 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml186_R1.fastq.gz
Encountered 9539 unique sequences from 13308 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml187_R1.fastq.gz
Encountered 16353 unique sequences from 24069 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml188_R1.fastq.gz
Encountered 18436 unique sequences from 30037 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml189_R1.fastq.gz
Encountered 15677 unique sequences from 23603 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml190_R1.fastq.gz
Encountered 5780 unique sequences from 7477 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml191_R1.fastq.gz
Encountered 29825 unique sequences from 44557 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml192_R1.fastq.gz
Encountered 47899 unique sequences from 78928 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml193_R1.fastq.gz
Encountered 24213 unique sequences from 39818 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml194_R1.fastq.gz
Encountered 41509 unique sequences from 69363 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml195_R1.fastq.gz
Encountered 35576 unique sequences from 60697 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml196_R1.fastq.gz
Encountered 65314 unique sequences from 121781 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml197_R1.fastq.gz
Encountered 30967 unique sequences from 49203 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml198_R1.fastq.gz
Encountered 20438 unique sequences from 31484 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml199_R1.fastq.gz
Encountered 34645 unique sequences from 57035 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml200_R1.fastq.gz
Encountered 126164 unique sequences from 262329 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml201_R1.fastq.gz
Encountered 121566 unique sequences from 257072 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml202_R1.fastq.gz
Encountered 20588 unique sequences from 31840 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml203_R1.fastq.gz
Encountered 72954 unique sequences from 135749 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml204_R1.fastq.gz
Encountered 58894 unique sequences from 105517 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml205_R1.fastq.gz
Encountered 36989 unique sequences from 67684 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml206_R1.fastq.gz
Encountered 28802 unique sequences from 51601 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml207_R1.fastq.gz
Encountered 28575 unique sequences from 46502 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml208_R1.fastq.gz
Encountered 9042 unique sequences from 13170 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml209_R1.fastq.gz
Encountered 13564 unique sequences from 19689 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml210_R1.fastq.gz
Encountered 16199 unique sequences from 24819 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml211_R1.fastq.gz
Encountered 14030 unique sequences from 21629 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml212_R1.fastq.gz
Encountered 16286 unique sequences from 25271 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml213_R1.fastq.gz
Encountered 13362 unique sequences from 20462 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml214_R1.fastq.gz
Encountered 19103 unique sequences from 27741 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml215_R1.fastq.gz
Encountered 18696 unique sequences from 26981 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml216_R1.fastq.gz
Encountered 119 unique sequences from 124 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml217_R1.fastq.gz
Encountered 1102 unique sequences from 1347 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml218_R1.fastq.gz
Encountered 8146 unique sequences from 11631 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml219_R1.fastq.gz
Encountered 10127 unique sequences from 15354 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml220_R1.fastq.gz
Encountered 8377 unique sequences from 11942 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml221_R1.fastq.gz
Encountered 5249 unique sequences from 6993 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml222_R1.fastq.gz
Encountered 14497 unique sequences from 21307 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml223_R1.fastq.gz
Encountered 21094 unique sequences from 32241 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml224_R1.fastq.gz
Encountered 443 unique sequences from 505 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml225_R1.fastq.gz
Encountered 12413 unique sequences from 19181 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml226_R1.fastq.gz
Encountered 7236 unique sequences from 9552 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml227_R1.fastq.gz
Encountered 11299 unique sequences from 16060 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml228_R1.fastq.gz
Encountered 15819 unique sequences from 23689 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml229_R1.fastq.gz
Encountered 10312 unique sequences from 15186 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml230_R1.fastq.gz
Encountered 29625 unique sequences from 50356 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml231_R1.fastq.gz
Encountered 28636 unique sequences from 46571 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml232_R1.fastq.gz
Encountered 11304 unique sequences from 16195 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml233_R1.fastq.gz
Encountered 11818 unique sequences from 17387 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml234_R1.fastq.gz
Encountered 8175 unique sequences from 11119 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml235_R1.fastq.gz
Encountered 11624 unique sequences from 17339 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml236_R1.fastq.gz
Encountered 9235 unique sequences from 13704 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml237_R1.fastq.gz
Encountered 6892 unique sequences from 9608 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml238_R1.fastq.gz
Encountered 6242 unique sequences from 8057 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml239_R1.fastq.gz
Encountered 4098 unique sequences from 5064 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml240_R1.fastq.gz
Encountered 9499 unique sequences from 13041 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml241_R1.fastq.gz
Encountered 19403 unique sequences from 31709 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml242_R1.fastq.gz
Encountered 12685 unique sequences from 19382 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml243_R1.fastq.gz
Encountered 8928 unique sequences from 13031 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml244_R1.fastq.gz
Encountered 6374 unique sequences from 8649 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml245_R1.fastq.gz
Encountered 8894 unique sequences from 12333 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml246_R1.fastq.gz
Encountered 110 unique sequences from 113 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml247_R1.fastq.gz
Encountered 18083 unique sequences from 27521 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml248_R1.fastq.gz
Encountered 3561 unique sequences from 4704 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml249_R1.fastq.gz
Encountered 16824 unique sequences from 25607 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml250_R1.fastq.gz
Encountered 16455 unique sequences from 23945 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml251_R1.fastq.gz
Encountered 14469 unique sequences from 21349 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml252_R1.fastq.gz
Encountered 9421 unique sequences from 13063 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml253_R1.fastq.gz
Encountered 3808 unique sequences from 5231 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml254_R1.fastq.gz
Encountered 12939 unique sequences from 20255 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml255_R1.fastq.gz
Encountered 16026 unique sequences from 24460 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml256_R1.fastq.gz
Encountered 10641 unique sequences from 14583 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml257_R1.fastq.gz
Encountered 8980 unique sequences from 13150 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml258_R1.fastq.gz
Encountered 10900 unique sequences from 15304 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml259_R1.fastq.gz
Encountered 12414 unique sequences from 18461 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml260_R1.fastq.gz
Encountered 7733 unique sequences from 10805 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml261_R1.fastq.gz
Encountered 4903 unique sequences from 6571 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml262_R1.fastq.gz
Encountered 8946 unique sequences from 12203 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml263_R1.fastq.gz
Encountered 905 unique sequences from 1017 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml264_R1.fastq.gz
Encountered 7032 unique sequences from 9461 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml265_R1.fastq.gz
Encountered 37198 unique sequences from 73681 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml266_R1.fastq.gz
Encountered 11365 unique sequences from 17674 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml267_R1.fastq.gz
Encountered 37940 unique sequences from 68938 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml268_R1.fastq.gz
Encountered 90 unique sequences from 97 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml269_R1.fastq.gz
Encountered 8369 unique sequences from 12665 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml270_R1.fastq.gz
Encountered 24871 unique sequences from 40554 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml271_R1.fastq.gz
Encountered 54877 unique sequences from 93260 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml272_R1.fastq.gz
Encountered 68884 unique sequences from 127107 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml273_R1.fastq.gz
Encountered 67262 unique sequences from 128219 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml274_R1.fastq.gz
Encountered 62758 unique sequences from 111146 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml275_R1.fastq.gz
Encountered 46309 unique sequences from 83661 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml276_R1.fastq.gz
Encountered 19404 unique sequences from 29864 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml277_R1.fastq.gz
Encountered 59959 unique sequences from 116807 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml278_R1.fastq.gz
Encountered 47234 unique sequences from 92530 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml279_R1.fastq.gz
Encountered 82792 unique sequences from 151236 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml280_R1.fastq.gz
Encountered 14461 unique sequences from 21836 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml281_R1.fastq.gz
Encountered 15899 unique sequences from 24598 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml282_R1.fastq.gz
Encountered 22459 unique sequences from 35393 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml283_R1.fastq.gz
Encountered 104305 unique sequences from 216060 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml284_R1.fastq.gz
Encountered 10092 unique sequences from 15396 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml285_R1.fastq.gz
Encountered 92667 unique sequences from 177479 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml286_R1.fastq.gz
Encountered 74864 unique sequences from 124107 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml287_R1.fastq.gz
Encountered 65482 unique sequences from 106466 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml288_R1.fastq.gz
Encountered 22854 unique sequences from 36902 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml289_R1.fastq.gz
Encountered 51659 unique sequences from 84835 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml290_R1.fastq.gz
Encountered 35696 unique sequences from 64847 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml291_R1.fastq.gz
Encountered 50131 unique sequences from 83552 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml292_R1.fastq.gz
Encountered 19392 unique sequences from 31066 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml293_R1.fastq.gz
Encountered 6280 unique sequences from 9995 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml294_R1.fastq.gz
Encountered 44179 unique sequences from 74230 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml295_R1.fastq.gz
Encountered 39005 unique sequences from 63637 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml296_R1.fastq.gz
Encountered 16955 unique sequences from 24781 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml297_R1.fastq.gz
Encountered 22645 unique sequences from 32742 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml298_R1.fastq.gz
Encountered 37751 unique sequences from 56715 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml299_R1.fastq.gz
Encountered 50013 unique sequences from 79202 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml300_R1.fastq.gz
Encountered 12329 unique sequences from 17006 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml301_R1.fastq.gz
Encountered 44820 unique sequences from 71273 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml302_R1.fastq.gz
Encountered 48704 unique sequences from 82160 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml303_R1.fastq.gz
Encountered 44837 unique sequences from 72905 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml304_R1.fastq.gz
Encountered 29634 unique sequences from 44463 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml305_R1.fastq.gz
Encountered 24035 unique sequences from 32232 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml306_R1.fastq.gz
Encountered 30568 unique sequences from 46392 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml307_R1.fastq.gz
Encountered 30013 unique sequences from 47335 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml308_R1.fastq.gz
Encountered 28195 unique sequences from 41845 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml309_R1.fastq.gz
Encountered 33250 unique sequences from 54867 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml310_R1.fastq.gz
Encountered 13639 unique sequences from 21072 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml311_R1.fastq.gz
Encountered 9637 unique sequences from 13707 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml312_R1.fastq.gz
Encountered 21476 unique sequences from 32967 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml001_R2.fastq.gz
Encountered 14957 unique sequences from 28541 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml002_R2.fastq.gz
Encountered 15266 unique sequences from 28011 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml003_R2.fastq.gz
Encountered 18174 unique sequences from 32988 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml004_R2.fastq.gz
Encountered 15081 unique sequences from 26821 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml005_R2.fastq.gz
Encountered 16653 unique sequences from 29697 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml006_R2.fastq.gz
Encountered 14175 unique sequences from 24584 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml007_R2.fastq.gz
Encountered 19465 unique sequences from 35704 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml008_R2.fastq.gz
Encountered 13729 unique sequences from 23478 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml009_R2.fastq.gz
Encountered 15846 unique sequences from 29377 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml010_R2.fastq.gz
Encountered 11419 unique sequences from 20444 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml011_R2.fastq.gz
Encountered 9273 unique sequences from 15374 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml012_R2.fastq.gz
Encountered 16908 unique sequences from 30208 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml013_R2.fastq.gz
Encountered 19732 unique sequences from 38160 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml014_R2.fastq.gz
Encountered 16018 unique sequences from 31418 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml015_R2.fastq.gz
Encountered 16225 unique sequences from 31670 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml016_R2.fastq.gz
Encountered 16048 unique sequences from 29259 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml017_R2.fastq.gz
Encountered 18342 unique sequences from 32921 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml018_R2.fastq.gz
Encountered 7888 unique sequences from 12627 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml019_R2.fastq.gz
Encountered 15403 unique sequences from 24680 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml020_R2.fastq.gz
Encountered 17755 unique sequences from 28375 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml021_R2.fastq.gz
Encountered 27528 unique sequences from 45727 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml022_R2.fastq.gz
Encountered 16851 unique sequences from 26948 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml023_R2.fastq.gz
Encountered 22079 unique sequences from 36563 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml024_R2.fastq.gz
Encountered 16779 unique sequences from 27677 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml025_R2.fastq.gz
Encountered 22328 unique sequences from 42203 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml026_R2.fastq.gz
Encountered 18256 unique sequences from 34263 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml027_R2.fastq.gz
Encountered 13385 unique sequences from 17086 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml028_R2.fastq.gz
Encountered 12586 unique sequences from 20692 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml029_R2.fastq.gz
Encountered 19502 unique sequences from 20355 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml030_R2.fastq.gz
Encountered 14629 unique sequences from 28067 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml031_R2.fastq.gz
Encountered 17761 unique sequences from 32440 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml032_R2.fastq.gz
Encountered 13176 unique sequences from 22011 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml033_R2.fastq.gz
Encountered 19128 unique sequences from 35035 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml034_R2.fastq.gz
Encountered 15986 unique sequences from 28303 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml035_R2.fastq.gz
Encountered 22376 unique sequences from 37176 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml036_R2.fastq.gz
Encountered 11810 unique sequences from 21730 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml037_R2.fastq.gz
Encountered 35416 unique sequences from 62880 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml038_R2.fastq.gz
Encountered 19138 unique sequences from 36838 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml039_R2.fastq.gz
Encountered 27585 unique sequences from 53845 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml040_R2.fastq.gz
Encountered 17202 unique sequences from 28445 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml041_R2.fastq.gz
Encountered 14715 unique sequences from 26284 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml042_R2.fastq.gz
Encountered 14176 unique sequences from 25614 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml043_R2.fastq.gz
Encountered 18883 unique sequences from 32664 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml044_R2.fastq.gz
Encountered 20630 unique sequences from 37410 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml045_R2.fastq.gz
Encountered 18505 unique sequences from 32629 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml046_R2.fastq.gz
Encountered 17070 unique sequences from 29876 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml047_R2.fastq.gz
Encountered 16517 unique sequences from 28506 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml048_R2.fastq.gz
Encountered 11568 unique sequences from 18819 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml049_R2.fastq.gz
Encountered 21410 unique sequences from 39117 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml050_R2.fastq.gz
Encountered 20336 unique sequences from 39383 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml051_R2.fastq.gz
Encountered 17515 unique sequences from 31202 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml052_R2.fastq.gz
Encountered 14199 unique sequences from 23204 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml053_R2.fastq.gz
Encountered 23333 unique sequences from 43062 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml054_R2.fastq.gz
Encountered 11483 unique sequences from 18867 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml055_R2.fastq.gz
Encountered 20453 unique sequences from 38385 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml056_R2.fastq.gz
Encountered 18832 unique sequences from 34664 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml057_R2.fastq.gz
Encountered 18808 unique sequences from 31887 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml058_R2.fastq.gz
Encountered 9464 unique sequences from 14906 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml059_R2.fastq.gz
Encountered 8829 unique sequences from 13525 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml060_R2.fastq.gz
Encountered 14656 unique sequences from 26375 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml061_R2.fastq.gz
Encountered 15536 unique sequences from 25137 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml062_R2.fastq.gz
Encountered 14834 unique sequences from 24128 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml063_R2.fastq.gz
Encountered 14432 unique sequences from 23847 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml064_R2.fastq.gz
Encountered 9782 unique sequences from 15571 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml065_R2.fastq.gz
Encountered 10361 unique sequences from 16496 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml066_R2.fastq.gz
Encountered 8204 unique sequences from 12605 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml067_R2.fastq.gz
Encountered 12807 unique sequences from 21047 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml068_R2.fastq.gz
Encountered 9162 unique sequences from 13995 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml069_R2.fastq.gz
Encountered 15174 unique sequences from 24812 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml070_R2.fastq.gz
Encountered 12824 unique sequences from 21059 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml071_R2.fastq.gz
Encountered 10206 unique sequences from 15942 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml072_R2.fastq.gz
Encountered 8375 unique sequences from 12651 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml073_R2.fastq.gz
Encountered 26157 unique sequences from 48494 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml074_R2.fastq.gz
Encountered 10801 unique sequences from 16040 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml075_R2.fastq.gz
Encountered 12828 unique sequences from 20835 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml076_R2.fastq.gz
Encountered 11873 unique sequences from 19162 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml077_R2.fastq.gz
Encountered 20130 unique sequences from 37252 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml078_R2.fastq.gz
Encountered 6186 unique sequences from 9819 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml079_R2.fastq.gz
Encountered 18096 unique sequences from 33832 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml080_R2.fastq.gz
Encountered 7778 unique sequences from 11979 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml081_R2.fastq.gz
Encountered 12279 unique sequences from 20150 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml082_R2.fastq.gz
Encountered 6586 unique sequences from 10707 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml083_R2.fastq.gz
Encountered 14984 unique sequences from 26175 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml084_R2.fastq.gz
Encountered 14900 unique sequences from 25757 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml085_R2.fastq.gz
Encountered 15231 unique sequences from 26498 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml086_R2.fastq.gz
Encountered 14643 unique sequences from 27627 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml087_R2.fastq.gz
Encountered 7053 unique sequences from 11186 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml088_R2.fastq.gz
Encountered 8513 unique sequences from 13337 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml089_R2.fastq.gz
Encountered 44564 unique sequences from 85392 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml090_R2.fastq.gz
Encountered 7962 unique sequences from 12134 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml091_R2.fastq.gz
Encountered 12794 unique sequences from 20730 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml092_R2.fastq.gz
Encountered 13826 unique sequences from 23491 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml093_R2.fastq.gz
Encountered 12130 unique sequences from 19559 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml094_R2.fastq.gz
Encountered 14706 unique sequences from 25711 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml095_R2.fastq.gz
Encountered 11563 unique sequences from 20894 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml096_R2.fastq.gz
Encountered 15943 unique sequences from 25959 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml097_R2.fastq.gz
Encountered 20610 unique sequences from 36632 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml098_R2.fastq.gz
Encountered 27130 unique sequences from 51569 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml099_R2.fastq.gz
Encountered 20775 unique sequences from 38561 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml100_R2.fastq.gz
Encountered 11628 unique sequences from 19340 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml101_R2.fastq.gz
Encountered 10771 unique sequences from 18156 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml102_R2.fastq.gz
Encountered 11864 unique sequences from 19193 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml103_R2.fastq.gz
Encountered 32925 unique sequences from 62024 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml104_R2.fastq.gz
Encountered 18301 unique sequences from 31177 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml105_R2.fastq.gz
Encountered 10068 unique sequences from 16338 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml106_R2.fastq.gz
Encountered 12472 unique sequences from 20974 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml107_R2.fastq.gz
Encountered 7114 unique sequences from 10761 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml108_R2.fastq.gz
Encountered 10702 unique sequences from 17267 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml109_R2.fastq.gz
Encountered 10077 unique sequences from 17517 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml110_R2.fastq.gz
Encountered 9718 unique sequences from 14812 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml111_R2.fastq.gz
Encountered 16606 unique sequences from 20561 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml112_R2.fastq.gz
Encountered 22612 unique sequences from 41817 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml113_R2.fastq.gz
Encountered 9329 unique sequences from 15211 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml114_R2.fastq.gz
Encountered 11793 unique sequences from 18839 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml115_R2.fastq.gz
Encountered 9339 unique sequences from 14752 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml116_R2.fastq.gz
Encountered 13131 unique sequences from 21980 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml117_R2.fastq.gz
Encountered 11049 unique sequences from 18694 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml118_R2.fastq.gz
Encountered 9688 unique sequences from 15305 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml119_R2.fastq.gz
Encountered 12921 unique sequences from 22531 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml120_R2.fastq.gz
Encountered 10600 unique sequences from 19170 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml121_R2.fastq.gz
Encountered 12065 unique sequences from 20736 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml122_R2.fastq.gz
Encountered 10564 unique sequences from 17611 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml123_R2.fastq.gz
Encountered 10630 unique sequences from 17226 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml124_R2.fastq.gz
Encountered 15125 unique sequences from 27681 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml125_R2.fastq.gz
Encountered 3300 unique sequences from 4927 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml126_R2.fastq.gz
Encountered 6923 unique sequences from 10783 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml127_R2.fastq.gz
Encountered 23046 unique sequences from 39001 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml128_R2.fastq.gz
Encountered 16718 unique sequences from 29604 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml129_R2.fastq.gz
Encountered 9441 unique sequences from 15395 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml130_R2.fastq.gz
Encountered 9752 unique sequences from 15946 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml131_R2.fastq.gz
Encountered 9513 unique sequences from 14535 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml132_R2.fastq.gz
Encountered 13380 unique sequences from 20130 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml133_R2.fastq.gz
Encountered 6772 unique sequences from 10486 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml134_R2.fastq.gz
Encountered 36811 unique sequences from 72813 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml135_R2.fastq.gz
Encountered 12705 unique sequences from 21767 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml136_R2.fastq.gz
Encountered 6864 unique sequences from 10371 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml137_R2.fastq.gz
Encountered 14739 unique sequences from 24798 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml138_R2.fastq.gz
Encountered 10750 unique sequences from 16445 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml139_R2.fastq.gz
Encountered 12196 unique sequences from 18935 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml140_R2.fastq.gz
Encountered 14370 unique sequences from 24416 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml141_R2.fastq.gz
Encountered 10125 unique sequences from 16289 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml142_R2.fastq.gz
Encountered 8857 unique sequences from 13046 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml143_R2.fastq.gz
Encountered 8709 unique sequences from 13182 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml144_R2.fastq.gz
Encountered 6452 unique sequences from 9963 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml145_R2.fastq.gz
Encountered 10212 unique sequences from 17346 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml146_R2.fastq.gz
Encountered 13154 unique sequences from 21719 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml147_R2.fastq.gz
Encountered 10410 unique sequences from 17394 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml148_R2.fastq.gz
Encountered 7136 unique sequences from 11488 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml149_R2.fastq.gz
Encountered 9195 unique sequences from 15029 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml150_R2.fastq.gz
Encountered 10640 unique sequences from 17264 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml151_R2.fastq.gz
Encountered 13672 unique sequences from 23194 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml152_R2.fastq.gz
Encountered 23591 unique sequences from 41708 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml153_R2.fastq.gz
Encountered 27756 unique sequences from 49950 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml154_R2.fastq.gz
Encountered 17962 unique sequences from 31188 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml155_R2.fastq.gz
Encountered 19904 unique sequences from 32055 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml156_R2.fastq.gz
Encountered 20171 unique sequences from 32118 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml157_R2.fastq.gz
Encountered 22780 unique sequences from 40929 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml158_R2.fastq.gz
Encountered 21449 unique sequences from 43054 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml159_R2.fastq.gz
Encountered 13759 unique sequences from 22894 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml160_R2.fastq.gz
Encountered 11223 unique sequences from 19298 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml161_R2.fastq.gz
Encountered 13357 unique sequences from 21161 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml162_R2.fastq.gz
Encountered 10457 unique sequences from 16489 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml163_R2.fastq.gz
Encountered 10928 unique sequences from 18268 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml164_R2.fastq.gz
Encountered 13408 unique sequences from 22554 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml165_R2.fastq.gz
Encountered 48 unique sequences from 51 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml166_R2.fastq.gz
Encountered 8758 unique sequences from 13707 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml167_R2.fastq.gz
Encountered 11308 unique sequences from 18048 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml168_R2.fastq.gz
Encountered 3912 unique sequences from 5974 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml169_R2.fastq.gz
Encountered 11337 unique sequences from 19347 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml170_R2.fastq.gz
Encountered 14046 unique sequences from 24081 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml171_R2.fastq.gz
Encountered 58 unique sequences from 63 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml172_R2.fastq.gz
Encountered 9902 unique sequences from 16107 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml173_R2.fastq.gz
Encountered 6680 unique sequences from 10923 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml174_R2.fastq.gz
Encountered 8849 unique sequences from 14282 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml175_R2.fastq.gz
Encountered 6837 unique sequences from 11231 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml176_R2.fastq.gz
Encountered 10817 unique sequences from 17328 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml177_R2.fastq.gz
Encountered 11083 unique sequences from 18237 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml178_R2.fastq.gz
Encountered 10110 unique sequences from 17043 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml179_R2.fastq.gz
Encountered 7892 unique sequences from 12042 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml180_R2.fastq.gz
Encountered 8939 unique sequences from 14477 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml181_R2.fastq.gz
Encountered 6611 unique sequences from 10550 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml182_R2.fastq.gz
Encountered 8165 unique sequences from 13384 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml183_R2.fastq.gz
Encountered 10449 unique sequences from 17422 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml184_R2.fastq.gz
Encountered 9603 unique sequences from 15542 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml185_R2.fastq.gz
Encountered 9096 unique sequences from 14755 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml186_R2.fastq.gz
Encountered 8879 unique sequences from 13308 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml187_R2.fastq.gz
Encountered 13816 unique sequences from 24069 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml188_R2.fastq.gz
Encountered 17564 unique sequences from 30037 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml189_R2.fastq.gz
Encountered 12450 unique sequences from 23603 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml190_R2.fastq.gz
Encountered 4893 unique sequences from 7477 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml191_R2.fastq.gz
Encountered 27326 unique sequences from 44557 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml192_R2.fastq.gz
Encountered 39833 unique sequences from 78928 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml193_R2.fastq.gz
Encountered 21350 unique sequences from 39818 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml194_R2.fastq.gz
Encountered 36667 unique sequences from 69363 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml195_R2.fastq.gz
Encountered 31551 unique sequences from 60697 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml196_R2.fastq.gz
Encountered 55715 unique sequences from 121781 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml197_R2.fastq.gz
Encountered 28215 unique sequences from 49203 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml198_R2.fastq.gz
Encountered 17978 unique sequences from 31484 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml199_R2.fastq.gz
Encountered 28358 unique sequences from 57035 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml200_R2.fastq.gz
Encountered 110098 unique sequences from 262329 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml201_R2.fastq.gz
Encountered 110314 unique sequences from 257072 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml202_R2.fastq.gz
Encountered 17439 unique sequences from 31840 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml203_R2.fastq.gz
Encountered 62552 unique sequences from 135749 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml204_R2.fastq.gz
Encountered 50710 unique sequences from 105517 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml205_R2.fastq.gz
Encountered 32503 unique sequences from 67684 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml206_R2.fastq.gz
Encountered 25787 unique sequences from 51601 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml207_R2.fastq.gz
Encountered 25361 unique sequences from 46502 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml208_R2.fastq.gz
Encountered 8221 unique sequences from 13170 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml209_R2.fastq.gz
Encountered 12299 unique sequences from 19689 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml210_R2.fastq.gz
Encountered 14861 unique sequences from 24819 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml211_R2.fastq.gz
Encountered 12790 unique sequences from 21629 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml212_R2.fastq.gz
Encountered 14974 unique sequences from 25271 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml213_R2.fastq.gz
Encountered 12181 unique sequences from 20462 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml214_R2.fastq.gz
Encountered 17154 unique sequences from 27741 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml215_R2.fastq.gz
Encountered 14652 unique sequences from 26981 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml216_R2.fastq.gz
Encountered 118 unique sequences from 124 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml217_R2.fastq.gz
Encountered 959 unique sequences from 1347 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml218_R2.fastq.gz
Encountered 7150 unique sequences from 11631 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml219_R2.fastq.gz
Encountered 8575 unique sequences from 15354 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml220_R2.fastq.gz
Encountered 6909 unique sequences from 11942 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml221_R2.fastq.gz
Encountered 4323 unique sequences from 6993 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml222_R2.fastq.gz
Encountered 12073 unique sequences from 21307 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml223_R2.fastq.gz
Encountered 16819 unique sequences from 32241 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml224_R2.fastq.gz
Encountered 403 unique sequences from 505 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml225_R2.fastq.gz
Encountered 11288 unique sequences from 19181 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml226_R2.fastq.gz
Encountered 6153 unique sequences from 9552 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml227_R2.fastq.gz
Encountered 9459 unique sequences from 16060 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml228_R2.fastq.gz
Encountered 12673 unique sequences from 23689 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml229_R2.fastq.gz
Encountered 8609 unique sequences from 15186 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml230_R2.fastq.gz
Encountered 24923 unique sequences from 50356 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml231_R2.fastq.gz
Encountered 27520 unique sequences from 46571 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml232_R2.fastq.gz
Encountered 9508 unique sequences from 16195 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml233_R2.fastq.gz
Encountered 10347 unique sequences from 17387 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml234_R2.fastq.gz
Encountered 7106 unique sequences from 11119 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml235_R2.fastq.gz
Encountered 10354 unique sequences from 17339 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml236_R2.fastq.gz
Encountered 7943 unique sequences from 13704 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml237_R2.fastq.gz
Encountered 6076 unique sequences from 9608 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml238_R2.fastq.gz
Encountered 5238 unique sequences from 8057 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml239_R2.fastq.gz
Encountered 3394 unique sequences from 5064 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml240_R2.fastq.gz
Encountered 7655 unique sequences from 13041 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml241_R2.fastq.gz
Encountered 16248 unique sequences from 31709 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml242_R2.fastq.gz
Encountered 10669 unique sequences from 19382 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml243_R2.fastq.gz
Encountered 7586 unique sequences from 13031 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml244_R2.fastq.gz
Encountered 5316 unique sequences from 8649 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml245_R2.fastq.gz
Encountered 7365 unique sequences from 12333 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml246_R2.fastq.gz
Encountered 109 unique sequences from 113 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml247_R2.fastq.gz
Encountered 15238 unique sequences from 27521 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml248_R2.fastq.gz
Encountered 3141 unique sequences from 4704 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml249_R2.fastq.gz
Encountered 14711 unique sequences from 25607 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml250_R2.fastq.gz
Encountered 14060 unique sequences from 23945 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml251_R2.fastq.gz
Encountered 12208 unique sequences from 21349 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml252_R2.fastq.gz
Encountered 8022 unique sequences from 13063 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml253_R2.fastq.gz
Encountered 3379 unique sequences from 5231 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml254_R2.fastq.gz
Encountered 11308 unique sequences from 20255 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml255_R2.fastq.gz
Encountered 13270 unique sequences from 24460 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml256_R2.fastq.gz
Encountered 9035 unique sequences from 14583 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml257_R2.fastq.gz
Encountered 7568 unique sequences from 13150 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml258_R2.fastq.gz
Encountered 9341 unique sequences from 15304 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml259_R2.fastq.gz
Encountered 10638 unique sequences from 18461 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml260_R2.fastq.gz
Encountered 6584 unique sequences from 10805 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml261_R2.fastq.gz
Encountered 4158 unique sequences from 6571 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml262_R2.fastq.gz
Encountered 7449 unique sequences from 12203 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml263_R2.fastq.gz
Encountered 800 unique sequences from 1017 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml264_R2.fastq.gz
Encountered 5852 unique sequences from 9461 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml265_R2.fastq.gz
Encountered 33792 unique sequences from 73681 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml266_R2.fastq.gz
Encountered 10045 unique sequences from 17674 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml267_R2.fastq.gz
Encountered 33982 unique sequences from 68938 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml268_R2.fastq.gz
Encountered 87 unique sequences from 97 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml269_R2.fastq.gz
Encountered 7076 unique sequences from 12665 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml270_R2.fastq.gz
Encountered 21933 unique sequences from 40554 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml271_R2.fastq.gz
Encountered 45402 unique sequences from 93260 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml272_R2.fastq.gz
Encountered 57991 unique sequences from 127107 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml273_R2.fastq.gz
Encountered 59204 unique sequences from 128219 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml274_R2.fastq.gz
Encountered 52392 unique sequences from 111146 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml275_R2.fastq.gz
Encountered 38862 unique sequences from 83661 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml276_R2.fastq.gz
Encountered 16114 unique sequences from 29864 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml277_R2.fastq.gz
Encountered 47534 unique sequences from 116807 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml278_R2.fastq.gz
Encountered 40467 unique sequences from 92530 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml279_R2.fastq.gz
Encountered 67940 unique sequences from 151236 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml280_R2.fastq.gz
Encountered 11866 unique sequences from 21836 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml281_R2.fastq.gz
Encountered 13372 unique sequences from 24598 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml282_R2.fastq.gz
Encountered 18978 unique sequences from 35393 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml283_R2.fastq.gz
Encountered 90557 unique sequences from 216060 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml284_R2.fastq.gz
Encountered 8338 unique sequences from 15396 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml285_R2.fastq.gz
Encountered 78237 unique sequences from 177479 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml286_R2.fastq.gz
Encountered 58857 unique sequences from 124107 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml287_R2.fastq.gz
Encountered 54687 unique sequences from 106466 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml288_R2.fastq.gz
Encountered 19056 unique sequences from 36902 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml289_R2.fastq.gz
Encountered 44654 unique sequences from 84835 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml290_R2.fastq.gz
Encountered 29531 unique sequences from 64847 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml291_R2.fastq.gz
Encountered 42818 unique sequences from 83552 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml292_R2.fastq.gz
Encountered 15956 unique sequences from 31066 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml293_R2.fastq.gz
Encountered 5317 unique sequences from 9995 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml294_R2.fastq.gz
Encountered 35447 unique sequences from 74230 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml295_R2.fastq.gz
Encountered 38382 unique sequences from 63637 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml296_R2.fastq.gz
Encountered 14851 unique sequences from 24781 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml297_R2.fastq.gz
Encountered 19789 unique sequences from 32742 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml298_R2.fastq.gz
Encountered 32078 unique sequences from 56715 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml299_R2.fastq.gz
Encountered 42294 unique sequences from 79202 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml300_R2.fastq.gz
Encountered 10325 unique sequences from 17006 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml301_R2.fastq.gz
Encountered 39801 unique sequences from 71273 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml302_R2.fastq.gz
Encountered 40311 unique sequences from 82160 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml303_R2.fastq.gz
Encountered 38400 unique sequences from 72905 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml304_R2.fastq.gz
Encountered 26200 unique sequences from 44463 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml305_R2.fastq.gz
Encountered 21190 unique sequences from 32232 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml306_R2.fastq.gz
Encountered 26806 unique sequences from 46392 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml307_R2.fastq.gz
Encountered 26410 unique sequences from 47335 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml308_R2.fastq.gz
Encountered 24633 unique sequences from 41845 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml309_R2.fastq.gz
Encountered 31086 unique sequences from 54867 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml310_R2.fastq.gz
Encountered 10736 unique sequences from 21072 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml311_R2.fastq.gz
Encountered 7810 unique sequences from 13707 total sequences read.
Dereplicating sequence entries in Fastq file: RAW/cutadapt/filtered/nml312_R2.fastq.gz
Encountered 18136 unique sequences from 32967 total sequences read.
</code></pre>
</details>

<br/>

# 3. DADA2 & ASV Inference

At this point we are ready to apply the core sample inference algorithm (dada) to the filtered and trimmed sequence data. DADA2 offers three options for whether and how to pool samples for ASV inference. 

If `pool = TRUE`, the algorithm will pool together all samples prior to sample inference.  
If `pool = FALSE`, sample inference is performed on each sample individually.  
If `pool = "pseudo"`, the algorithm will perform pseudo-pooling between individually processed samples.

For our final analysis, we chose `pool = pseudo` for this data set. 


```{r}
dadaFs <- dada(derepFs, err = errF, pool = "pseudo", multithread = TRUE)
```

<details markdown="1">
<summary>Detailed results of `dada` on forward reads</summary>
<pre><code>
Sample 1 - 28541 reads in 17193 unique sequences.
Sample 2 - 28011 reads in 17850 unique sequences.
Sample 3 - 32988 reads in 20482 unique sequences.
Sample 4 - 26821 reads in 17661 unique sequences.
Sample 5 - 29697 reads in 19229 unique sequences.
Sample 6 - 24584 reads in 16925 unique sequences.
Sample 7 - 35704 reads in 22571 unique sequences.
Sample 8 - 23478 reads in 15941 unique sequences.
Sample 9 - 29377 reads in 18508 unique sequences.
Sample 10 - 20444 reads in 13916 unique sequences.
Sample 11 - 15374 reads in 10725 unique sequences.
Sample 12 - 30208 reads in 20125 unique sequences.
Sample 13 - 38160 reads in 21931 unique sequences.
Sample 14 - 31418 reads in 19443 unique sequences.
Sample 15 - 31670 reads in 18851 unique sequences.
Sample 16 - 29259 reads in 19173 unique sequences.
Sample 17 - 32921 reads in 20708 unique sequences.
Sample 18 - 12627 reads in 8896 unique sequences.
Sample 19 - 24680 reads in 16682 unique sequences.
Sample 20 - 28375 reads in 18934 unique sequences.
Sample 21 - 45727 reads in 29363 unique sequences.
Sample 22 - 26948 reads in 19032 unique sequences.
Sample 23 - 36563 reads in 24450 unique sequences.
Sample 24 - 27677 reads in 18963 unique sequences.
Sample 25 - 42203 reads in 25904 unique sequences.
Sample 26 - 34263 reads in 21719 unique sequences.
Sample 27 - 17086 reads in 11782 unique sequences.
Sample 28 - 20692 reads in 14699 unique sequences.
Sample 29 - 20355 reads in 13560 unique sequences.
Sample 30 - 28067 reads in 18514 unique sequences.
Sample 31 - 32440 reads in 21336 unique sequences.
Sample 32 - 22011 reads in 15621 unique sequences.
Sample 33 - 35035 reads in 22780 unique sequences.
Sample 34 - 28303 reads in 19534 unique sequences.
Sample 35 - 37176 reads in 24412 unique sequences.
Sample 36 - 21730 reads in 14666 unique sequences.
Sample 37 - 62880 reads in 37349 unique sequences.
Sample 38 - 36838 reads in 23549 unique sequences.
Sample 39 - 53845 reads in 32398 unique sequences.
Sample 40 - 28445 reads in 19416 unique sequences.
Sample 41 - 26284 reads in 17310 unique sequences.
Sample 42 - 25614 reads in 17057 unique sequences.
Sample 43 - 32664 reads in 21265 unique sequences.
Sample 44 - 37410 reads in 23885 unique sequences.
Sample 45 - 32629 reads in 21188 unique sequences.
Sample 46 - 29876 reads in 20747 unique sequences.
Sample 47 - 28506 reads in 19428 unique sequences.
Sample 48 - 18819 reads in 13609 unique sequences.
Sample 49 - 39117 reads in 24937 unique sequences.
Sample 50 - 39383 reads in 24574 unique sequences.
Sample 51 - 31202 reads in 20126 unique sequences.
Sample 52 - 23204 reads in 16424 unique sequences.
Sample 53 - 43062 reads in 27550 unique sequences.
Sample 54 - 18867 reads in 13869 unique sequences.
Sample 55 - 38385 reads in 24471 unique sequences.
Sample 56 - 34664 reads in 22869 unique sequences.
Sample 57 - 31887 reads in 21597 unique sequences.
Sample 58 - 14906 reads in 11071 unique sequences.
Sample 59 - 13525 reads in 9966 unique sequences.
Sample 60 - 26375 reads in 17519 unique sequences.
Sample 61 - 25137 reads in 17105 unique sequences.
Sample 62 - 24128 reads in 17099 unique sequences.
Sample 63 - 23847 reads in 16225 unique sequences.
Sample 64 - 15571 reads in 11290 unique sequences.
Sample 65 - 16496 reads in 11921 unique sequences.
Sample 66 - 12605 reads in 9231 unique sequences.
Sample 67 - 21047 reads in 14594 unique sequences.
Sample 68 - 13995 reads in 10362 unique sequences.
Sample 69 - 24812 reads in 17298 unique sequences.
Sample 70 - 21059 reads in 15618 unique sequences.
Sample 71 - 15942 reads in 12070 unique sequences.
Sample 72 - 12651 reads in 9946 unique sequences.
Sample 73 - 48494 reads in 29360 unique sequences.
Sample 74 - 16040 reads in 11150 unique sequences.
Sample 75 - 20835 reads in 14078 unique sequences.
Sample 76 - 19162 reads in 13389 unique sequences.
Sample 77 - 37252 reads in 23023 unique sequences.
Sample 78 - 9819 reads in 7245 unique sequences.
Sample 79 - 33832 reads in 20826 unique sequences.
Sample 80 - 11979 reads in 8409 unique sequences.
Sample 81 - 20150 reads in 13982 unique sequences.
Sample 82 - 10707 reads in 7936 unique sequences.
Sample 83 - 26175 reads in 17380 unique sequences.
Sample 84 - 25757 reads in 17689 unique sequences.
Sample 85 - 26498 reads in 16886 unique sequences.
Sample 86 - 27627 reads in 17887 unique sequences.
Sample 87 - 11186 reads in 8087 unique sequences.
Sample 88 - 13337 reads in 9834 unique sequences.
Sample 89 - 85392 reads in 50969 unique sequences.
Sample 90 - 12134 reads in 9120 unique sequences.
Sample 91 - 20730 reads in 14209 unique sequences.
Sample 92 - 23491 reads in 15786 unique sequences.
Sample 93 - 19559 reads in 13674 unique sequences.
Sample 94 - 25711 reads in 17937 unique sequences.
Sample 95 - 20894 reads in 14026 unique sequences.
Sample 96 - 25959 reads in 18415 unique sequences.
Sample 97 - 36632 reads in 24008 unique sequences.
Sample 98 - 51569 reads in 30864 unique sequences.
Sample 99 - 38561 reads in 24304 unique sequences.
Sample 100 - 19340 reads in 13126 unique sequences.
Sample 101 - 18156 reads in 13019 unique sequences.
Sample 102 - 19193 reads in 13669 unique sequences.
Sample 103 - 62024 reads in 39168 unique sequences.
Sample 104 - 31177 reads in 21063 unique sequences.
Sample 105 - 16338 reads in 11560 unique sequences.
Sample 106 - 20974 reads in 14683 unique sequences.
Sample 107 - 10761 reads in 8326 unique sequences.
Sample 108 - 17267 reads in 12568 unique sequences.
Sample 109 - 17517 reads in 12086 unique sequences.
Sample 110 - 14812 reads in 10792 unique sequences.
Sample 111 - 20561 reads in 14724 unique sequences.
Sample 112 - 41817 reads in 26233 unique sequences.
Sample 113 - 15211 reads in 10895 unique sequences.
Sample 114 - 18839 reads in 13098 unique sequences.
Sample 115 - 14752 reads in 10647 unique sequences.
Sample 116 - 21980 reads in 14545 unique sequences.
Sample 117 - 18694 reads in 13079 unique sequences.
Sample 118 - 15305 reads in 11169 unique sequences.
Sample 119 - 22531 reads in 15666 unique sequences.
Sample 120 - 19170 reads in 13085 unique sequences.
Sample 121 - 20736 reads in 14259 unique sequences.
Sample 122 - 17611 reads in 12051 unique sequences.
Sample 123 - 17226 reads in 12196 unique sequences.
Sample 124 - 27681 reads in 17926 unique sequences.
Sample 125 - 4927 reads in 3899 unique sequences.
Sample 126 - 10783 reads in 7955 unique sequences.
Sample 127 - 39001 reads in 26408 unique sequences.
Sample 128 - 29604 reads in 19690 unique sequences.
Sample 129 - 15395 reads in 11142 unique sequences.
Sample 130 - 15946 reads in 11379 unique sequences.
Sample 131 - 14535 reads in 11190 unique sequences.
Sample 132 - 20130 reads in 14990 unique sequences.
Sample 133 - 10486 reads in 7965 unique sequences.
Sample 134 - 72813 reads in 43443 unique sequences.
Sample 135 - 21767 reads in 15026 unique sequences.
Sample 136 - 10371 reads in 7761 unique sequences.
Sample 137 - 24798 reads in 16890 unique sequences.
Sample 138 - 16445 reads in 11762 unique sequences.
Sample 139 - 18935 reads in 13472 unique sequences.
Sample 140 - 24416 reads in 16007 unique sequences.
Sample 141 - 16289 reads in 11855 unique sequences.
Sample 142 - 13046 reads in 10114 unique sequences.
Sample 143 - 13182 reads in 10199 unique sequences.
Sample 144 - 9963 reads in 7598 unique sequences.
Sample 145 - 17346 reads in 11906 unique sequences.
Sample 146 - 21719 reads in 14971 unique sequences.
Sample 147 - 17394 reads in 12014 unique sequences.
Sample 148 - 11488 reads in 8248 unique sequences.
Sample 149 - 15029 reads in 11106 unique sequences.
Sample 150 - 17264 reads in 12343 unique sequences.
Sample 151 - 23194 reads in 16265 unique sequences.
Sample 152 - 41708 reads in 26754 unique sequences.
Sample 153 - 49950 reads in 31821 unique sequences.
Sample 154 - 31188 reads in 20740 unique sequences.
Sample 155 - 32055 reads in 22684 unique sequences.
Sample 156 - 32118 reads in 22631 unique sequences.
Sample 157 - 40929 reads in 26865 unique sequences.
Sample 158 - 43054 reads in 24828 unique sequences.
Sample 159 - 22894 reads in 15718 unique sequences.
Sample 160 - 19298 reads in 13191 unique sequences.
Sample 161 - 21161 reads in 15170 unique sequences.
Sample 162 - 16489 reads in 11782 unique sequences.
Sample 163 - 18268 reads in 12616 unique sequences.
Sample 164 - 22554 reads in 14949 unique sequences.
Sample 165 - 51 reads in 45 unique sequences.
Sample 166 - 13707 reads in 10107 unique sequences.
Sample 167 - 18048 reads in 13414 unique sequences.
Sample 168 - 5974 reads in 4647 unique sequences.
Sample 169 - 19347 reads in 13176 unique sequences.
Sample 170 - 24081 reads in 15928 unique sequences.
Sample 171 - 63 reads in 56 unique sequences.
Sample 172 - 16107 reads in 11289 unique sequences.
Sample 173 - 10923 reads in 8220 unique sequences.
Sample 174 - 14282 reads in 10418 unique sequences.
Sample 175 - 11231 reads in 8143 unique sequences.
Sample 176 - 17328 reads in 12384 unique sequences.
Sample 177 - 18237 reads in 13000 unique sequences.
Sample 178 - 17043 reads in 11923 unique sequences.
Sample 179 - 12042 reads in 9327 unique sequences.
Sample 180 - 14477 reads in 10400 unique sequences.
Sample 181 - 10550 reads in 7802 unique sequences.
Sample 182 - 13384 reads in 9448 unique sequences.
Sample 183 - 17422 reads in 12034 unique sequences.
Sample 184 - 15542 reads in 11133 unique sequences.
Sample 185 - 14755 reads in 10672 unique sequences.
Sample 186 - 13308 reads in 9539 unique sequences.
Sample 187 - 24069 reads in 16353 unique sequences.
Sample 188 - 30037 reads in 18436 unique sequences.
Sample 189 - 23603 reads in 15677 unique sequences.
Sample 190 - 7477 reads in 5780 unique sequences.
Sample 191 - 44557 reads in 29825 unique sequences.
Sample 192 - 78928 reads in 47899 unique sequences.
Sample 193 - 39818 reads in 24213 unique sequences.
Sample 194 - 69363 reads in 41509 unique sequences.
Sample 195 - 60697 reads in 35576 unique sequences.
Sample 196 - 121781 reads in 65314 unique sequences.
Sample 197 - 49203 reads in 30967 unique sequences.
Sample 198 - 31484 reads in 20438 unique sequences.
Sample 199 - 57035 reads in 34645 unique sequences.
Sample 200 - 262329 reads in 126164 unique sequences.
Sample 201 - 257072 reads in 121566 unique sequences.
Sample 202 - 31840 reads in 20588 unique sequences.
Sample 203 - 135749 reads in 72954 unique sequences.
Sample 204 - 105517 reads in 58894 unique sequences.
Sample 205 - 67684 reads in 36989 unique sequences.
Sample 206 - 51601 reads in 28802 unique sequences.
Sample 207 - 46502 reads in 28575 unique sequences.
Sample 208 - 13170 reads in 9042 unique sequences.
Sample 209 - 19689 reads in 13564 unique sequences.
Sample 210 - 24819 reads in 16199 unique sequences.
Sample 211 - 21629 reads in 14030 unique sequences.
Sample 212 - 25271 reads in 16286 unique sequences.
Sample 213 - 20462 reads in 13362 unique sequences.
Sample 214 - 27741 reads in 19103 unique sequences.
Sample 215 - 26981 reads in 18696 unique sequences.
Sample 216 - 124 reads in 119 unique sequences.
Sample 217 - 1347 reads in 1102 unique sequences.
Sample 218 - 11631 reads in 8146 unique sequences.
Sample 219 - 15354 reads in 10127 unique sequences.
Sample 220 - 11942 reads in 8377 unique sequences.
Sample 221 - 6993 reads in 5249 unique sequences.
Sample 222 - 21307 reads in 14497 unique sequences.
Sample 223 - 32241 reads in 21094 unique sequences.
Sample 224 - 505 reads in 443 unique sequences.
Sample 225 - 19181 reads in 12413 unique sequences.
Sample 226 - 9552 reads in 7236 unique sequences.
Sample 227 - 16060 reads in 11299 unique sequences.
Sample 228 - 23689 reads in 15819 unique sequences.
Sample 229 - 15186 reads in 10312 unique sequences.
Sample 230 - 50356 reads in 29625 unique sequences.
Sample 231 - 46571 reads in 28636 unique sequences.
Sample 232 - 16195 reads in 11304 unique sequences.
Sample 233 - 17387 reads in 11818 unique sequences.
Sample 234 - 11119 reads in 8175 unique sequences.
Sample 235 - 17339 reads in 11624 unique sequences.
Sample 236 - 13704 reads in 9235 unique sequences.
Sample 237 - 9608 reads in 6892 unique sequences.
Sample 238 - 8057 reads in 6242 unique sequences.
Sample 239 - 5064 reads in 4098 unique sequences.
Sample 240 - 13041 reads in 9499 unique sequences.
Sample 241 - 31709 reads in 19403 unique sequences.
Sample 242 - 19382 reads in 12685 unique sequences.
Sample 243 - 13031 reads in 8928 unique sequences.
Sample 244 - 8649 reads in 6374 unique sequences.
Sample 245 - 12333 reads in 8894 unique sequences.
Sample 246 - 113 reads in 110 unique sequences.
Sample 247 - 27521 reads in 18083 unique sequences.
Sample 248 - 4704 reads in 3561 unique sequences.
Sample 249 - 25607 reads in 16824 unique sequences.
Sample 250 - 23945 reads in 16455 unique sequences.
Sample 251 - 21349 reads in 14469 unique sequences.
Sample 252 - 13063 reads in 9421 unique sequences.
Sample 253 - 5231 reads in 3808 unique sequences.
Sample 254 - 20255 reads in 12939 unique sequences.
Sample 255 - 24460 reads in 16026 unique sequences.
Sample 256 - 14583 reads in 10641 unique sequences.
Sample 257 - 13150 reads in 8980 unique sequences.
Sample 258 - 15304 reads in 10900 unique sequences.
Sample 259 - 18461 reads in 12414 unique sequences.
Sample 260 - 10805 reads in 7733 unique sequences.
Sample 261 - 6571 reads in 4903 unique sequences.
Sample 262 - 12203 reads in 8946 unique sequences.
Sample 263 - 1017 reads in 905 unique sequences.
Sample 264 - 9461 reads in 7032 unique sequences.
Sample 265 - 73681 reads in 37198 unique sequences.
Sample 266 - 17674 reads in 11365 unique sequences.
Sample 267 - 68938 reads in 37940 unique sequences.
Sample 268 - 97 reads in 90 unique sequences.
Sample 269 - 12665 reads in 8369 unique sequences.
Sample 270 - 40554 reads in 24871 unique sequences.
Sample 271 - 93260 reads in 54877 unique sequences.
Sample 272 - 127107 reads in 68884 unique sequences.
Sample 273 - 128219 reads in 67262 unique sequences.
Sample 274 - 111146 reads in 62758 unique sequences.
Sample 275 - 83661 reads in 46309 unique sequences.
Sample 276 - 29864 reads in 19404 unique sequences.
Sample 277 - 116807 reads in 59959 unique sequences.
Sample 278 - 92530 reads in 47234 unique sequences.
Sample 279 - 151236 reads in 82792 unique sequences.
Sample 280 - 21836 reads in 14461 unique sequences.
Sample 281 - 24598 reads in 15899 unique sequences.
Sample 282 - 35393 reads in 22459 unique sequences.
Sample 283 - 216060 reads in 104305 unique sequences.
Sample 284 - 15396 reads in 10092 unique sequences.
Sample 285 - 177479 reads in 92667 unique sequences.
Sample 286 - 124107 reads in 74864 unique sequences.
Sample 287 - 106466 reads in 65482 unique sequences.
Sample 288 - 36902 reads in 22854 unique sequences.
Sample 289 - 84835 reads in 51659 unique sequences.
Sample 290 - 64847 reads in 35696 unique sequences.
Sample 291 - 83552 reads in 50131 unique sequences.
Sample 292 - 31066 reads in 19392 unique sequences.
Sample 293 - 9995 reads in 6280 unique sequences.
Sample 294 - 74230 reads in 44179 unique sequences.
Sample 295 - 63637 reads in 39005 unique sequences.
Sample 296 - 24781 reads in 16955 unique sequences.
Sample 297 - 32742 reads in 22645 unique sequences.
Sample 298 - 56715 reads in 37751 unique sequences.
Sample 299 - 79202 reads in 50013 unique sequences.
Sample 300 - 17006 reads in 12329 unique sequences.
Sample 301 - 71273 reads in 44820 unique sequences.
Sample 302 - 82160 reads in 48704 unique sequences.
Sample 303 - 72905 reads in 44837 unique sequences.
Sample 304 - 44463 reads in 29634 unique sequences.
Sample 305 - 32232 reads in 24035 unique sequences.
Sample 306 - 46392 reads in 30568 unique sequences.
Sample 307 - 47335 reads in 30013 unique sequences.
Sample 308 - 41845 reads in 28195 unique sequences.
Sample 309 - 54867 reads in 33250 unique sequences.
Sample 310 - 21072 reads in 13639 unique sequences.
Sample 311 - 13707 reads in 9637 unique sequences.
Sample 312 - 32967 reads in 21476 unique sequences.
</code></pre>
</details>

As an example, we can inspect the returned `dada-class` object for the forward and reverse reads from the sample #`r sampnum`:

```{r}
dadaFs[[2]]
```

```
dada-class: object describing DADA2 denoising results
3783 sequence variants were inferred from 17850 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16
```

This output tells us how many true sequence variants the DADA2 algorithm inferred from the  unique sequences, in this case the sample 2.

```{r}
dadaRs <- dada(derepRs, err = errR, pool = "pseudo", multithread = TRUE)
```

<details markdown="1">
<summary>Detailed results of `dada` on reverse reads</summary>
<pre><code>
Sample 1 - 28541 reads in 14957 unique sequences.
Sample 2 - 28011 reads in 15266 unique sequences.
Sample 3 - 32988 reads in 18174 unique sequences.
Sample 4 - 26821 reads in 15081 unique sequences.
Sample 5 - 29697 reads in 16653 unique sequences.
Sample 6 - 24584 reads in 14175 unique sequences.
Sample 7 - 35704 reads in 19465 unique sequences.
Sample 8 - 23478 reads in 13729 unique sequences.
Sample 9 - 29377 reads in 15846 unique sequences.
Sample 10 - 20444 reads in 11419 unique sequences.
Sample 11 - 15374 reads in 9273 unique sequences.
Sample 12 - 30208 reads in 16908 unique sequences.
Sample 13 - 38160 reads in 19732 unique sequences.
Sample 14 - 31418 reads in 16018 unique sequences.
Sample 15 - 31670 reads in 16225 unique sequences.
Sample 16 - 29259 reads in 16048 unique sequences.
Sample 17 - 32921 reads in 18342 unique sequences.
Sample 18 - 12627 reads in 7888 unique sequences.
Sample 19 - 24680 reads in 15403 unique sequences.
Sample 20 - 28375 reads in 17755 unique sequences.
Sample 21 - 45727 reads in 27528 unique sequences.
Sample 22 - 26948 reads in 16851 unique sequences.
Sample 23 - 36563 reads in 22079 unique sequences.
Sample 24 - 27677 reads in 16779 unique sequences.
Sample 25 - 42203 reads in 22328 unique sequences.
Sample 26 - 34263 reads in 18256 unique sequences.
Sample 27 - 17086 reads in 13385 unique sequences.
Sample 28 - 20692 reads in 12586 unique sequences.
Sample 29 - 20355 reads in 19502 unique sequences.
Sample 30 - 28067 reads in 14629 unique sequences.
Sample 31 - 32440 reads in 17761 unique sequences.
Sample 32 - 22011 reads in 13176 unique sequences.
Sample 33 - 35035 reads in 19128 unique sequences.
Sample 34 - 28303 reads in 15986 unique sequences.
Sample 35 - 37176 reads in 22376 unique sequences.
Sample 36 - 21730 reads in 11810 unique sequences.
Sample 37 - 62880 reads in 35416 unique sequences.
Sample 38 - 36838 reads in 19138 unique sequences.
Sample 39 - 53845 reads in 27585 unique sequences.
Sample 40 - 28445 reads in 17202 unique sequences.
Sample 41 - 26284 reads in 14715 unique sequences.
Sample 42 - 25614 reads in 14176 unique sequences.
Sample 43 - 32664 reads in 18883 unique sequences.
Sample 44 - 37410 reads in 20630 unique sequences.
Sample 45 - 32629 reads in 18505 unique sequences.
Sample 46 - 29876 reads in 17070 unique sequences.
Sample 47 - 28506 reads in 16517 unique sequences.
Sample 48 - 18819 reads in 11568 unique sequences.
Sample 49 - 39117 reads in 21410 unique sequences.
Sample 50 - 39383 reads in 20336 unique sequences.
Sample 51 - 31202 reads in 17515 unique sequences.
Sample 52 - 23204 reads in 14199 unique sequences.
Sample 53 - 43062 reads in 23333 unique sequences.
Sample 54 - 18867 reads in 11483 unique sequences.
Sample 55 - 38385 reads in 20453 unique sequences.
Sample 56 - 34664 reads in 18832 unique sequences.
Sample 57 - 31887 reads in 18808 unique sequences.
Sample 58 - 14906 reads in 9464 unique sequences.
Sample 59 - 13525 reads in 8829 unique sequences.
Sample 60 - 26375 reads in 14656 unique sequences.
Sample 61 - 25137 reads in 15536 unique sequences.
Sample 62 - 24128 reads in 14834 unique sequences.
Sample 63 - 23847 reads in 14432 unique sequences.
Sample 64 - 15571 reads in 9782 unique sequences.
Sample 65 - 16496 reads in 10361 unique sequences.
Sample 66 - 12605 reads in 8204 unique sequences.
Sample 67 - 21047 reads in 12807 unique sequences.
Sample 68 - 13995 reads in 9162 unique sequences.
Sample 69 - 24812 reads in 15174 unique sequences.
Sample 70 - 21059 reads in 12824 unique sequences.
Sample 71 - 15942 reads in 10206 unique sequences.
Sample 72 - 12651 reads in 8375 unique sequences.
Sample 73 - 48494 reads in 26157 unique sequences.
Sample 74 - 16040 reads in 10801 unique sequences.
Sample 75 - 20835 reads in 12828 unique sequences.
Sample 76 - 19162 reads in 11873 unique sequences.
Sample 77 - 37252 reads in 20130 unique sequences.
Sample 78 - 9819 reads in 6186 unique sequences.
Sample 79 - 33832 reads in 18096 unique sequences.
Sample 80 - 11979 reads in 7778 unique sequences.
Sample 81 - 20150 reads in 12279 unique sequences.
Sample 82 - 10707 reads in 6586 unique sequences.
Sample 83 - 26175 reads in 14984 unique sequences.
Sample 84 - 25757 reads in 14900 unique sequences.
Sample 85 - 26498 reads in 15231 unique sequences.
Sample 86 - 27627 reads in 14643 unique sequences.
Sample 87 - 11186 reads in 7053 unique sequences.
Sample 88 - 13337 reads in 8513 unique sequences.
Sample 89 - 85392 reads in 44564 unique sequences.
Sample 90 - 12134 reads in 7962 unique sequences.
Sample 91 - 20730 reads in 12794 unique sequences.
Sample 92 - 23491 reads in 13826 unique sequences.
Sample 93 - 19559 reads in 12130 unique sequences.
Sample 94 - 25711 reads in 14706 unique sequences.
Sample 95 - 20894 reads in 11563 unique sequences.
Sample 96 - 25959 reads in 15943 unique sequences.
Sample 97 - 36632 reads in 20610 unique sequences.
Sample 98 - 51569 reads in 27130 unique sequences.
Sample 99 - 38561 reads in 20775 unique sequences.
Sample 100 - 19340 reads in 11628 unique sequences.
Sample 101 - 18156 reads in 10771 unique sequences.
Sample 102 - 19193 reads in 11864 unique sequences.
Sample 103 - 62024 reads in 32925 unique sequences.
Sample 104 - 31177 reads in 18301 unique sequences.
Sample 105 - 16338 reads in 10068 unique sequences.
Sample 106 - 20974 reads in 12472 unique sequences.
Sample 107 - 10761 reads in 7114 unique sequences.
Sample 108 - 17267 reads in 10702 unique sequences.
Sample 109 - 17517 reads in 10077 unique sequences.
Sample 110 - 14812 reads in 9718 unique sequences.
Sample 111 - 20561 reads in 16606 unique sequences.
Sample 112 - 41817 reads in 22612 unique sequences.
Sample 113 - 15211 reads in 9329 unique sequences.
Sample 114 - 18839 reads in 11793 unique sequences.
Sample 115 - 14752 reads in 9339 unique sequences.
Sample 116 - 21980 reads in 13131 unique sequences.
Sample 117 - 18694 reads in 11049 unique sequences.
Sample 118 - 15305 reads in 9688 unique sequences.
Sample 119 - 22531 reads in 12921 unique sequences.
Sample 120 - 19170 reads in 10600 unique sequences.
Sample 121 - 20736 reads in 12065 unique sequences.
Sample 122 - 17611 reads in 10564 unique sequences.
Sample 123 - 17226 reads in 10630 unique sequences.
Sample 124 - 27681 reads in 15125 unique sequences.
Sample 125 - 4927 reads in 3300 unique sequences.
Sample 126 - 10783 reads in 6923 unique sequences.
Sample 127 - 39001 reads in 23046 unique sequences.
Sample 128 - 29604 reads in 16718 unique sequences.
Sample 129 - 15395 reads in 9441 unique sequences.
Sample 130 - 15946 reads in 9752 unique sequences.
Sample 131 - 14535 reads in 9513 unique sequences.
Sample 132 - 20130 reads in 13380 unique sequences.
Sample 133 - 10486 reads in 6772 unique sequences.
Sample 134 - 72813 reads in 36811 unique sequences.
Sample 135 - 21767 reads in 12705 unique sequences.
Sample 136 - 10371 reads in 6864 unique sequences.
Sample 137 - 24798 reads in 14739 unique sequences.
Sample 138 - 16445 reads in 10750 unique sequences.
Sample 139 - 18935 reads in 12196 unique sequences.
Sample 140 - 24416 reads in 14370 unique sequences.
Sample 141 - 16289 reads in 10125 unique sequences.
Sample 142 - 13046 reads in 8857 unique sequences.
Sample 143 - 13182 reads in 8709 unique sequences.
Sample 144 - 9963 reads in 6452 unique sequences.
Sample 145 - 17346 reads in 10212 unique sequences.
Sample 146 - 21719 reads in 13154 unique sequences.
Sample 147 - 17394 reads in 10410 unique sequences.
Sample 148 - 11488 reads in 7136 unique sequences.
Sample 149 - 15029 reads in 9195 unique sequences.
Sample 150 - 17264 reads in 10640 unique sequences.
Sample 151 - 23194 reads in 13672 unique sequences.
Sample 152 - 41708 reads in 23591 unique sequences.
Sample 153 - 49950 reads in 27756 unique sequences.
Sample 154 - 31188 reads in 17962 unique sequences.
Sample 155 - 32055 reads in 19904 unique sequences.
Sample 156 - 32118 reads in 20171 unique sequences.
Sample 157 - 40929 reads in 22780 unique sequences.
Sample 158 - 43054 reads in 21449 unique sequences.
Sample 159 - 22894 reads in 13759 unique sequences.
Sample 160 - 19298 reads in 11223 unique sequences.
Sample 161 - 21161 reads in 13357 unique sequences.
Sample 162 - 16489 reads in 10457 unique sequences.
Sample 163 - 18268 reads in 10928 unique sequences.
Sample 164 - 22554 reads in 13408 unique sequences.
Sample 165 - 51 reads in 48 unique sequences.
Sample 166 - 13707 reads in 8758 unique sequences.
Sample 167 - 18048 reads in 11308 unique sequences.
Sample 168 - 5974 reads in 3912 unique sequences.
Sample 169 - 19347 reads in 11337 unique sequences.
Sample 170 - 24081 reads in 14046 unique sequences.
Sample 171 - 63 reads in 58 unique sequences.
Sample 172 - 16107 reads in 9902 unique sequences.
Sample 173 - 10923 reads in 6680 unique sequences.
Sample 174 - 14282 reads in 8849 unique sequences.
Sample 175 - 11231 reads in 6837 unique sequences.
Sample 176 - 17328 reads in 10817 unique sequences.
Sample 177 - 18237 reads in 11083 unique sequences.
Sample 178 - 17043 reads in 10110 unique sequences.
Sample 179 - 12042 reads in 7892 unique sequences.
Sample 180 - 14477 reads in 8939 unique sequences.
Sample 181 - 10550 reads in 6611 unique sequences.
Sample 182 - 13384 reads in 8165 unique sequences.
Sample 183 - 17422 reads in 10449 unique sequences.
Sample 184 - 15542 reads in 9603 unique sequences.
Sample 185 - 14755 reads in 9096 unique sequences.
Sample 186 - 13308 reads in 8879 unique sequences.
Sample 187 - 24069 reads in 13816 unique sequences.
Sample 188 - 30037 reads in 17564 unique sequences.
Sample 189 - 23603 reads in 12450 unique sequences.
Sample 190 - 7477 reads in 4893 unique sequences.
Sample 191 - 44557 reads in 27326 unique sequences.
Sample 192 - 78928 reads in 39833 unique sequences.
Sample 193 - 39818 reads in 21350 unique sequences.
Sample 194 - 69363 reads in 36667 unique sequences.
Sample 195 - 60697 reads in 31551 unique sequences.
Sample 196 - 121781 reads in 55715 unique sequences.
Sample 197 - 49203 reads in 28215 unique sequences.
Sample 198 - 31484 reads in 17978 unique sequences.
Sample 199 - 57035 reads in 28358 unique sequences.
Sample 200 - 262329 reads in 110098 unique sequences.
Sample 201 - 257072 reads in 110314 unique sequences.
Sample 202 - 31840 reads in 17439 unique sequences.
Sample 203 - 135749 reads in 62552 unique sequences.
Sample 204 - 105517 reads in 50710 unique sequences.
Sample 205 - 67684 reads in 32503 unique sequences.
Sample 206 - 51601 reads in 25787 unique sequences.
Sample 207 - 46502 reads in 25361 unique sequences.
Sample 208 - 13170 reads in 8221 unique sequences.
Sample 209 - 19689 reads in 12299 unique sequences.
Sample 210 - 24819 reads in 14861 unique sequences.
Sample 211 - 21629 reads in 12790 unique sequences.
Sample 212 - 25271 reads in 14974 unique sequences.
Sample 213 - 20462 reads in 12181 unique sequences.
Sample 214 - 27741 reads in 17154 unique sequences.
Sample 215 - 26981 reads in 14652 unique sequences.
Sample 216 - 124 reads in 118 unique sequences.
Sample 217 - 1347 reads in 959 unique sequences.
Sample 218 - 11631 reads in 7150 unique sequences.
Sample 219 - 15354 reads in 8575 unique sequences.
Sample 220 - 11942 reads in 6909 unique sequences.
Sample 221 - 6993 reads in 4323 unique sequences.
Sample 222 - 21307 reads in 12073 unique sequences.
Sample 223 - 32241 reads in 16819 unique sequences.
Sample 224 - 505 reads in 403 unique sequences.
Sample 225 - 19181 reads in 11288 unique sequences.
Sample 226 - 9552 reads in 6153 unique sequences.
Sample 227 - 16060 reads in 9459 unique sequences.
Sample 228 - 23689 reads in 12673 unique sequences.
Sample 229 - 15186 reads in 8609 unique sequences.
Sample 230 - 50356 reads in 24923 unique sequences.
Sample 231 - 46571 reads in 27520 unique sequences.
Sample 232 - 16195 reads in 9508 unique sequences.
Sample 233 - 17387 reads in 10347 unique sequences.
Sample 234 - 11119 reads in 7106 unique sequences.
Sample 235 - 17339 reads in 10354 unique sequences.
Sample 236 - 13704 reads in 7943 unique sequences.
Sample 237 - 9608 reads in 6076 unique sequences.
Sample 238 - 8057 reads in 5238 unique sequences.
Sample 239 - 5064 reads in 3394 unique sequences.
Sample 240 - 13041 reads in 7655 unique sequences.
Sample 241 - 31709 reads in 16248 unique sequences.
Sample 242 - 19382 reads in 10669 unique sequences.
Sample 243 - 13031 reads in 7586 unique sequences.
Sample 244 - 8649 reads in 5316 unique sequences.
Sample 245 - 12333 reads in 7365 unique sequences.
Sample 246 - 113 reads in 109 unique sequences.
Sample 247 - 27521 reads in 15238 unique sequences.
Sample 248 - 4704 reads in 3141 unique sequences.
Sample 249 - 25607 reads in 14711 unique sequences.
Sample 250 - 23945 reads in 14060 unique sequences.
Sample 251 - 21349 reads in 12208 unique sequences.
Sample 252 - 13063 reads in 8022 unique sequences.
Sample 253 - 5231 reads in 3379 unique sequences.
Sample 254 - 20255 reads in 11308 unique sequences.
Sample 255 - 24460 reads in 13270 unique sequences.
Sample 256 - 14583 reads in 9035 unique sequences.
Sample 257 - 13150 reads in 7568 unique sequences.
Sample 258 - 15304 reads in 9341 unique sequences.
Sample 259 - 18461 reads in 10638 unique sequences.
Sample 260 - 10805 reads in 6584 unique sequences.
Sample 261 - 6571 reads in 4158 unique sequences.
Sample 262 - 12203 reads in 7449 unique sequences.
Sample 263 - 1017 reads in 800 unique sequences.
Sample 264 - 9461 reads in 5852 unique sequences.
Sample 265 - 73681 reads in 33792 unique sequences.
Sample 266 - 17674 reads in 10045 unique sequences.
Sample 267 - 68938 reads in 33982 unique sequences.
Sample 268 - 97 reads in 87 unique sequences.
Sample 269 - 12665 reads in 7076 unique sequences.
Sample 270 - 40554 reads in 21933 unique sequences.
Sample 271 - 93260 reads in 45402 unique sequences.
Sample 272 - 127107 reads in 57991 unique sequences.
Sample 273 - 128219 reads in 59204 unique sequences.
Sample 274 - 111146 reads in 52392 unique sequences.
Sample 275 - 83661 reads in 38862 unique sequences.
Sample 276 - 29864 reads in 16114 unique sequences.
Sample 277 - 116807 reads in 47534 unique sequences.
Sample 278 - 92530 reads in 40467 unique sequences.
Sample 279 - 151236 reads in 67940 unique sequences.
Sample 280 - 21836 reads in 11866 unique sequences.
Sample 281 - 24598 reads in 13372 unique sequences.
Sample 282 - 35393 reads in 18978 unique sequences.
Sample 283 - 216060 reads in 90557 unique sequences.
Sample 284 - 15396 reads in 8338 unique sequences.
Sample 285 - 177479 reads in 78237 unique sequences.
Sample 286 - 124107 reads in 58857 unique sequences.
Sample 287 - 106466 reads in 54687 unique sequences.
Sample 288 - 36902 reads in 19056 unique sequences.
Sample 289 - 84835 reads in 44654 unique sequences.
Sample 290 - 64847 reads in 29531 unique sequences.
Sample 291 - 83552 reads in 42818 unique sequences.
Sample 292 - 31066 reads in 15956 unique sequences.
Sample 293 - 9995 reads in 5317 unique sequences.
Sample 294 - 74230 reads in 35447 unique sequences.
Sample 295 - 63637 reads in 38382 unique sequences.
Sample 296 - 24781 reads in 14851 unique sequences.
Sample 297 - 32742 reads in 19789 unique sequences.
Sample 298 - 56715 reads in 32078 unique sequences.
Sample 299 - 79202 reads in 42294 unique sequences.
Sample 300 - 17006 reads in 10325 unique sequences.
Sample 301 - 71273 reads in 39801 unique sequences.
Sample 302 - 82160 reads in 40311 unique sequences.
Sample 303 - 72905 reads in 38400 unique sequences.
Sample 304 - 44463 reads in 26200 unique sequences.
Sample 305 - 32232 reads in 21190 unique sequences.
Sample 306 - 46392 reads in 26806 unique sequences.
Sample 307 - 47335 reads in 26410 unique sequences.
Sample 308 - 41845 reads in 24633 unique sequences.
Sample 309 - 54867 reads in 31086 unique sequences.
Sample 310 - 21072 reads in 10736 unique sequences.
Sample 311 - 13707 reads in 7810 unique sequences.
Sample 312 - 32967 reads in 18136 unique sequences.
</code></pre>
</details>

As an example, we can inspect the returned `dada-class` object for the forward and reverse reads from the sample #`r sampnum`:

```{r}
dadaRs[[2]]
```

```
dada-class: object describing DADA2 denoising results
2802 sequence variants were inferred from 15266 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16
```

This output tells us how many true sequence variants the DADA2 algorithm inferred from the  unique sequences, in this case the sample 2.

<br/>

```{r, echo=FALSE, eval=TRUE}
denoise <- read.table("files/trepo/dada2/tables/ssu_denoise.txt",
                       header = TRUE, row.names = 1)
datatable(denoise, width = "80%",
          caption = htmltools::tags$caption(
            style = 'caption-side: bottom; text-align: left;',
            'Table: ', htmltools::em('Number of forward & reverse reads after denoising.')),
          extensions = 'Buttons', options = list(
            scrollX = TRUE,
            dom = 'Blfrtip',
            buttons = c('copy', 'csv', 'excel'),
            pageLength = 5,
            lengthMenu = list(c(5, 10, -1), c("5", "10", "All"))
            )
          )
```

# 4. Merge Paired Reads & Construct Sequence Tables

Now we construct an amplicon sequence variant (ASV) table.

```{r}
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs, verbose=TRUE, minOverlap = 20)
```

<details markdown="1">
<summary>Detailed results of merging reads</summary>
<pre><code>
22800 paired-reads (in 3726 unique pairings) successfully merged out of 27084 (in 6591 pairings) input.
21659 paired-reads (in 4222 unique pairings) successfully merged out of 26295 (in 7593 pairings) input.
26047 paired-reads (in 4820 unique pairings) successfully merged out of 31114 (in 8298 pairings) input.
20680 paired-reads (in 4241 unique pairings) successfully merged out of 24931 (in 7160 pairings) input.
23034 paired-reads (in 4485 unique pairings) successfully merged out of 27746 (in 7631 pairings) input.
18679 paired-reads (in 3914 unique pairings) successfully merged out of 22912 (in 6915 pairings) input.
27024 paired-reads (in 5088 unique pairings) successfully merged out of 33476 (in 9632 pairings) input.
17309 paired-reads (in 3899 unique pairings) successfully merged out of 21729 (in 7229 pairings) input.
23589 paired-reads (in 4582 unique pairings) successfully merged out of 27693 (in 7362 pairings) input.
16251 paired-reads (in 3679 unique pairings) successfully merged out of 18819 (in 5218 pairings) input.
11919 paired-reads (in 2964 unique pairings) successfully merged out of 14068 (in 4241 pairings) input.
23449 paired-reads (in 4642 unique pairings) successfully merged out of 27981 (in 7451 pairings) input.
31181 paired-reads (in 4691 unique pairings) successfully merged out of 36259 (in 8008 pairings) input.
25919 paired-reads (in 4300 unique pairings) successfully merged out of 29869 (in 6943 pairings) input.
25589 paired-reads (in 4267 unique pairings) successfully merged out of 29977 (in 7176 pairings) input.
23071 paired-reads (in 4686 unique pairings) successfully merged out of 27169 (in 7400 pairings) input.
26988 paired-reads (in 5051 unique pairings) successfully merged out of 30761 (in 7308 pairings) input.
9588 paired-reads (in 2510 unique pairings) successfully merged out of 11550 (in 3823 pairings) input.
16105 paired-reads (in 3689 unique pairings) successfully merged out of 22510 (in 8599 pairings) input.
19187 paired-reads (in 4188 unique pairings) successfully merged out of 26176 (in 9391 pairings) input.
29640 paired-reads (in 5650 unique pairings) successfully merged out of 41872 (in 14703 pairings) input.
18504 paired-reads (in 4244 unique pairings) successfully merged out of 24564 (in 8676 pairings) input.
24395 paired-reads (in 4994 unique pairings) successfully merged out of 33485 (in 11617 pairings) input.
20656 paired-reads (in 4447 unique pairings) successfully merged out of 25790 (in 8155 pairings) input.
30382 paired-reads (in 5259 unique pairings) successfully merged out of 39545 (in 12176 pairings) input.
25375 paired-reads (in 4860 unique pairings) successfully merged out of 32159 (in 9866 pairings) input.
11043 paired-reads (in 2665 unique pairings) successfully merged out of 15611 (in 6096 pairings) input.
13414 paired-reads (in 3284 unique pairings) successfully merged out of 18711 (in 7188 pairings) input.
11741 paired-reads (in 2066 unique pairings) successfully merged out of 17473 (in 5062 pairings) input.
22894 paired-reads (in 4400 unique pairings) successfully merged out of 26341 (in 6598 pairings) input.
24006 paired-reads (in 4862 unique pairings) successfully merged out of 30215 (in 9385 pairings) input.
15055 paired-reads (in 3803 unique pairings) successfully merged out of 20233 (in 7760 pairings) input.
25538 paired-reads (in 5152 unique pairings) successfully merged out of 32549 (in 10178 pairings) input.
19517 paired-reads (in 4420 unique pairings) successfully merged out of 25780 (in 8924 pairings) input.
26331 paired-reads (in 5086 unique pairings) successfully merged out of 34419 (in 10672 pairings) input.
17514 paired-reads (in 3785 unique pairings) successfully merged out of 20130 (in 5300 pairings) input.
45731 paired-reads (in 7047 unique pairings) successfully merged out of 58755 (in 16224 pairings) input.
27491 paired-reads (in 5054 unique pairings) successfully merged out of 34519 (in 10152 pairings) input.
41851 paired-reads (in 6643 unique pairings) successfully merged out of 50893 (in 12931 pairings) input.
19816 paired-reads (in 4268 unique pairings) successfully merged out of 26195 (in 8977 pairings) input.
20579 paired-reads (in 4310 unique pairings) successfully merged out of 24538 (in 7014 pairings) input.
19398 paired-reads (in 4161 unique pairings) successfully merged out of 23764 (in 7257 pairings) input.
23951 paired-reads (in 5009 unique pairings) successfully merged out of 30371 (in 9743 pairings) input.
28760 paired-reads (in 5322 unique pairings) successfully merged out of 35101 (in 9835 pairings) input.
24344 paired-reads (in 4806 unique pairings) successfully merged out of 30333 (in 9048 pairings) input.
21883 paired-reads (in 4558 unique pairings) successfully merged out of 27590 (in 8675 pairings) input.
19924 paired-reads (in 4344 unique pairings) successfully merged out of 26200 (in 8975 pairings) input.
13092 paired-reads (in 3362 unique pairings) successfully merged out of 17113 (in 6385 pairings) input.
27335 paired-reads (in 5172 unique pairings) successfully merged out of 36223 (in 11802 pairings) input.
29623 paired-reads (in 5234 unique pairings) successfully merged out of 36868 (in 10521 pairings) input.
21949 paired-reads (in 4466 unique pairings) successfully merged out of 28666 (in 9417 pairings) input.
14265 paired-reads (in 3490 unique pairings) successfully merged out of 20692 (in 8277 pairings) input.
30551 paired-reads (in 5414 unique pairings) successfully merged out of 39915 (in 11940 pairings) input.
11908 paired-reads (in 3116 unique pairings) successfully merged out of 16874 (in 6886 pairings) input.
28615 paired-reads (in 5607 unique pairings) successfully merged out of 35872 (in 10740 pairings) input.
25720 paired-reads (in 5214 unique pairings) successfully merged out of 32333 (in 10215 pairings) input.
20976 paired-reads (in 4596 unique pairings) successfully merged out of 29321 (in 10909 pairings) input.
9576 paired-reads (in 2715 unique pairings) successfully merged out of 13310 (in 5555 pairings) input.
8788 paired-reads (in 2561 unique pairings) successfully merged out of 12092 (in 4989 pairings) input.
21004 paired-reads (in 4261 unique pairings) successfully merged out of 24654 (in 6540 pairings) input.
15242 paired-reads (in 3620 unique pairings) successfully merged out of 22494 (in 9030 pairings) input.
15165 paired-reads (in 3545 unique pairings) successfully merged out of 21995 (in 8821 pairings) input.
14931 paired-reads (in 3627 unique pairings) successfully merged out of 21512 (in 8630 pairings) input.
10664 paired-reads (in 2866 unique pairings) successfully merged out of 14227 (in 5632 pairings) input.
10508 paired-reads (in 2849 unique pairings) successfully merged out of 14869 (in 6174 pairings) input.
8251 paired-reads (in 2382 unique pairings) successfully merged out of 11336 (in 4701 pairings) input.
14192 paired-reads (in 3514 unique pairings) successfully merged out of 19243 (in 7292 pairings) input.
8464 paired-reads (in 2413 unique pairings) successfully merged out of 12392 (in 5502 pairings) input.
15258 paired-reads (in 3555 unique pairings) successfully merged out of 22358 (in 8971 pairings) input.
14302 paired-reads (in 3560 unique pairings) successfully merged out of 19021 (in 6986 pairings) input.
9982 paired-reads (in 2852 unique pairings) successfully merged out of 14194 (in 6029 pairings) input.
7862 paired-reads (in 2405 unique pairings) successfully merged out of 11244 (in 5009 pairings) input.
35625 paired-reads (in 5846 unique pairings) successfully merged out of 45419 (in 12965 pairings) input.
10579 paired-reads (in 2577 unique pairings) successfully merged out of 14666 (in 5721 pairings) input.
13795 paired-reads (in 3336 unique pairings) successfully merged out of 18913 (in 7190 pairings) input.
13157 paired-reads (in 3257 unique pairings) successfully merged out of 17475 (in 6397 pairings) input.
29978 paired-reads (in 5245 unique pairings) successfully merged out of 34991 (in 8421 pairings) input.
7145 paired-reads (in 1937 unique pairings) successfully merged out of 8990 (in 3261 pairings) input.
26674 paired-reads (in 4608 unique pairings) successfully merged out of 31963 (in 8165 pairings) input.
9665 paired-reads (in 2503 unique pairings) successfully merged out of 11216 (in 3520 pairings) input.
13492 paired-reads (in 3336 unique pairings) successfully merged out of 18284 (in 6949 pairings) input.
8192 paired-reads (in 2375 unique pairings) successfully merged out of 9706 (in 3316 pairings) input.
20523 paired-reads (in 4411 unique pairings) successfully merged out of 24115 (in 6610 pairings) input.
19696 paired-reads (in 4381 unique pairings) successfully merged out of 23615 (in 6807 pairings) input.
19606 paired-reads (in 3977 unique pairings) successfully merged out of 24613 (in 7590 pairings) input.
21947 paired-reads (in 4184 unique pairings) successfully merged out of 26026 (in 6966 pairings) input.
7563 paired-reads (in 2179 unique pairings) successfully merged out of 10063 (in 4068 pairings) input.
8959 paired-reads (in 2534 unique pairings) successfully merged out of 12032 (in 4862 pairings) input.
59047 paired-reads (in 8233 unique pairings) successfully merged out of 79344 (in 22842 pairings) input.
7070 paired-reads (in 2138 unique pairings) successfully merged out of 10661 (in 4835 pairings) input.
13390 paired-reads (in 3304 unique pairings) successfully merged out of 18746 (in 7341 pairings) input.
15781 paired-reads (in 3583 unique pairings) successfully merged out of 21377 (in 7755 pairings) input.
11949 paired-reads (in 2974 unique pairings) successfully merged out of 17448 (in 7039 pairings) input.
19284 paired-reads (in 4129 unique pairings) successfully merged out of 23711 (in 7187 pairings) input.
16847 paired-reads (in 3753 unique pairings) successfully merged out of 19363 (in 5258 pairings) input.
16508 paired-reads (in 4029 unique pairings) successfully merged out of 23067 (in 8870 pairings) input.
23757 paired-reads (in 4900 unique pairings) successfully merged out of 33620 (in 12327 pairings) input.
37628 paired-reads (in 6154 unique pairings) successfully merged out of 48234 (in 13755 pairings) input.
28388 paired-reads (in 5227 unique pairings) successfully merged out of 35968 (in 10740 pairings) input.
12834 paired-reads (in 3017 unique pairings) successfully merged out of 17574 (in 6477 pairings) input.
11896 paired-reads (in 2976 unique pairings) successfully merged out of 16325 (in 6398 pairings) input.
11780 paired-reads (in 3100 unique pairings) successfully merged out of 17146 (in 7193 pairings) input.
42682 paired-reads (in 7110 unique pairings) successfully merged out of 57835 (in 18715 pairings) input.
20475 paired-reads (in 4506 unique pairings) successfully merged out of 28643 (in 10804 pairings) input.
10947 paired-reads (in 2793 unique pairings) successfully merged out of 14916 (in 5807 pairings) input.
15051 paired-reads (in 3604 unique pairings) successfully merged out of 19092 (in 6437 pairings) input.
7228 paired-reads (in 2156 unique pairings) successfully merged out of 9265 (in 3434 pairings) input.
11853 paired-reads (in 3100 unique pairings) successfully merged out of 15453 (in 5639 pairings) input.
11865 paired-reads (in 2688 unique pairings) successfully merged out of 16258 (in 5995 pairings) input.
9058 paired-reads (in 2532 unique pairings) successfully merged out of 13215 (in 5742 pairings) input.
12801 paired-reads (in 3021 unique pairings) successfully merged out of 18588 (in 7467 pairings) input.
30556 paired-reads (in 5574 unique pairings) successfully merged out of 38885 (in 11528 pairings) input.
10376 paired-reads (in 2699 unique pairings) successfully merged out of 13723 (in 5266 pairings) input.
12132 paired-reads (in 3063 unique pairings) successfully merged out of 16961 (in 6619 pairings) input.
9280 paired-reads (in 2484 unique pairings) successfully merged out of 13237 (in 5522 pairings) input.
15180 paired-reads (in 3280 unique pairings) successfully merged out of 20219 (in 6826 pairings) input.
13326 paired-reads (in 3187 unique pairings) successfully merged out of 17314 (in 6140 pairings) input.
11084 paired-reads (in 2875 unique pairings) successfully merged out of 13757 (in 4671 pairings) input.
17315 paired-reads (in 3975 unique pairings) successfully merged out of 20895 (in 6473 pairings) input.
15225 paired-reads (in 3542 unique pairings) successfully merged out of 17695 (in 5102 pairings) input.
13956 paired-reads (in 3346 unique pairings) successfully merged out of 18918 (in 7172 pairings) input.
11782 paired-reads (in 2938 unique pairings) successfully merged out of 16003 (in 6090 pairings) input.
11067 paired-reads (in 2822 unique pairings) successfully merged out of 15547 (in 6192 pairings) input.
20867 paired-reads (in 4201 unique pairings) successfully merged out of 25626 (in 7415 pairings) input.
3254 paired-reads (in 1158 unique pairings) successfully merged out of 4345 (in 1962 pairings) input.
6960 paired-reads (in 2142 unique pairings) successfully merged out of 9593 (in 4104 pairings) input.
25463 paired-reads (in 5449 unique pairings) successfully merged out of 35851 (in 13618 pairings) input.
20390 paired-reads (in 4436 unique pairings) successfully merged out of 27314 (in 9623 pairings) input.
10294 paired-reads (in 2794 unique pairings) successfully merged out of 13869 (in 5529 pairings) input.
11838 paired-reads (in 3168 unique pairings) successfully merged out of 14686 (in 5115 pairings) input.
8624 paired-reads (in 2514 unique pairings) successfully merged out of 12787 (in 5686 pairings) input.
11417 paired-reads (in 3134 unique pairings) successfully merged out of 17297 (in 7501 pairings) input.
6646 paired-reads (in 2076 unique pairings) successfully merged out of 9362 (in 4231 pairings) input.
50375 paired-reads (in 7768 unique pairings) successfully merged out of 68027 (in 20825 pairings) input.
14677 paired-reads (in 3345 unique pairings) successfully merged out of 19971 (in 7455 pairings) input.
6748 paired-reads (in 2023 unique pairings) successfully merged out of 9194 (in 3861 pairings) input.
16782 paired-reads (in 3758 unique pairings) successfully merged out of 22701 (in 8209 pairings) input.
10120 paired-reads (in 2681 unique pairings) successfully merged out of 14662 (in 6116 pairings) input.
11980 paired-reads (in 3131 unique pairings) successfully merged out of 17028 (in 7057 pairings) input.
17673 paired-reads (in 3759 unique pairings) successfully merged out of 22627 (in 7338 pairings) input.
10521 paired-reads (in 2759 unique pairings) successfully merged out of 14783 (in 6117 pairings) input.
7539 paired-reads (in 2281 unique pairings) successfully merged out of 11258 (in 5069 pairings) input.
8807 paired-reads (in 2637 unique pairings) successfully merged out of 11707 (in 4744 pairings) input.
6718 paired-reads (in 2104 unique pairings) successfully merged out of 8751 (in 3570 pairings) input.
12120 paired-reads (in 2943 unique pairings) successfully merged out of 15857 (in 5773 pairings) input.
13890 paired-reads (in 3411 unique pairings) successfully merged out of 19587 (in 7751 pairings) input.
11809 paired-reads (in 2915 unique pairings) successfully merged out of 15915 (in 5979 pairings) input.
7699 paired-reads (in 2087 unique pairings) successfully merged out of 10260 (in 4008 pairings) input.
9581 paired-reads (in 2650 unique pairings) successfully merged out of 13453 (in 5610 pairings) input.
10613 paired-reads (in 2835 unique pairings) successfully merged out of 15385 (in 6444 pairings) input.
15592 paired-reads (in 3734 unique pairings) successfully merged out of 21322 (in 8109 pairings) input.
28835 paired-reads (in 5502 unique pairings) successfully merged out of 38776 (in 12852 pairings) input.
34000 paired-reads (in 6363 unique pairings) successfully merged out of 46379 (in 15583 pairings) input.
22623 paired-reads (in 4759 unique pairings) successfully merged out of 28721 (in 9003 pairings) input.
19462 paired-reads (in 4527 unique pairings) successfully merged out of 28589 (in 11265 pairings) input.
19144 paired-reads (in 4427 unique pairings) successfully merged out of 28467 (in 11207 pairings) input.
27920 paired-reads (in 5324 unique pairings) successfully merged out of 37670 (in 12710 pairings) input.
34011 paired-reads (in 5109 unique pairings) successfully merged out of 40724 (in 9674 pairings) input.
15418 paired-reads (in 3712 unique pairings) successfully merged out of 21032 (in 8059 pairings) input.
14239 paired-reads (in 3269 unique pairings) successfully merged out of 17739 (in 5756 pairings) input.
12481 paired-reads (in 3260 unique pairings) successfully merged out of 18795 (in 8076 pairings) input.
10318 paired-reads (in 2691 unique pairings) successfully merged out of 14765 (in 6064 pairings) input.
12485 paired-reads (in 2909 unique pairings) successfully merged out of 16713 (in 5928 pairings) input.
15535 paired-reads (in 3373 unique pairings) successfully merged out of 20779 (in 7152 pairings) input.
4 paired-reads (in 2 unique pairings) successfully merged out of 14 (in 4 pairings) input.
8873 paired-reads (in 2548 unique pairings) successfully merged out of 12214 (in 4962 pairings) input.
12283 paired-reads (in 3262 unique pairings) successfully merged out of 16238 (in 6216 pairings) input.
4353 paired-reads (in 1427 unique pairings) successfully merged out of 5202 (in 1978 pairings) input.
13070 paired-reads (in 3119 unique pairings) successfully merged out of 17767 (in 6646 pairings) input.
16052 paired-reads (in 3618 unique pairings) successfully merged out of 22109 (in 8179 pairings) input.
7 paired-reads (in 4 unique pairings) successfully merged out of 15 (in 6 pairings) input.
10489 paired-reads (in 2681 unique pairings) successfully merged out of 14569 (in 5689 pairings) input.
7368 paired-reads (in 2113 unique pairings) successfully merged out of 9791 (in 3889 pairings) input.
8881 paired-reads (in 2498 unique pairings) successfully merged out of 12730 (in 5368 pairings) input.
8355 paired-reads (in 2442 unique pairings) successfully merged out of 10347 (in 3890 pairings) input.
11248 paired-reads (in 3129 unique pairings) successfully merged out of 15479 (in 6282 pairings) input.
11629 paired-reads (in 3074 unique pairings) successfully merged out of 16423 (in 6746 pairings) input.
12725 paired-reads (in 3161 unique pairings) successfully merged out of 15591 (in 5143 pairings) input.
8338 paired-reads (in 2296 unique pairings) successfully merged out of 10773 (in 3910 pairings) input.
10129 paired-reads (in 2749 unique pairings) successfully merged out of 13110 (in 4813 pairings) input.
7212 paired-reads (in 2082 unique pairings) successfully merged out of 9623 (in 3921 pairings) input.
9095 paired-reads (in 2466 unique pairings) successfully merged out of 12168 (in 4793 pairings) input.
12130 paired-reads (in 3013 unique pairings) successfully merged out of 16121 (in 6116 pairings) input.
10247 paired-reads (in 2790 unique pairings) successfully merged out of 13840 (in 5512 pairings) input.
9985 paired-reads (in 2681 unique pairings) successfully merged out of 13249 (in 5072 pairings) input.
8954 paired-reads (in 2498 unique pairings) successfully merged out of 11894 (in 4621 pairings) input.
17092 paired-reads (in 3863 unique pairings) successfully merged out of 22133 (in 7482 pairings) input.
23926 paired-reads (in 3960 unique pairings) successfully merged out of 28192 (in 6597 pairings) input.
18698 paired-reads (in 3766 unique pairings) successfully merged out of 21992 (in 5928 pairings) input.
4981 paired-reads (in 1618 unique pairings) successfully merged out of 6568 (in 2745 pairings) input.
31593 paired-reads (in 5858 unique pairings) successfully merged out of 41328 (in 12951 pairings) input.
59381 paired-reads (in 8270 unique pairings) successfully merged out of 73798 (in 18008 pairings) input.
30267 paired-reads (in 5177 unique pairings) successfully merged out of 37477 (in 10257 pairings) input.
51466 paired-reads (in 7715 unique pairings) successfully merged out of 65554 (in 18000 pairings) input.
45008 paired-reads (in 6857 unique pairings) successfully merged out of 56997 (in 15540 pairings) input.
97014 paired-reads (in 9885 unique pairings) successfully merged out of 115599 (in 21663 pairings) input.
35847 paired-reads (in 5838 unique pairings) successfully merged out of 46133 (in 13434 pairings) input.
22872 paired-reads (in 4527 unique pairings) successfully merged out of 29271 (in 9183 pairings) input.
42973 paired-reads (in 5804 unique pairings) successfully merged out of 53720 (in 13230 pairings) input.
204167 paired-reads (in 13388 unique pairings) successfully merged out of 249653 (in 42413 pairings) input.
201009 paired-reads (in 13389 unique pairings) successfully merged out of 244494 (in 40801 pairings) input.
24721 paired-reads (in 4698 unique pairings) successfully merged out of 29720 (in 7865 pairings) input.
104033 paired-reads (in 10282 unique pairings) successfully merged out of 127118 (in 24413 pairings) input.
80007 paired-reads (in 9000 unique pairings) successfully merged out of 98854 (in 20570 pairings) input.
54141 paired-reads (in 6986 unique pairings) successfully merged out of 64647 (in 14471 pairings) input.
41880 paired-reads (in 6011 unique pairings) successfully merged out of 49210 (in 11249 pairings) input.
35689 paired-reads (in 5762 unique pairings) successfully merged out of 43935 (in 11862 pairings) input.
9723 paired-reads (in 2540 unique pairings) successfully merged out of 12166 (in 4287 pairings) input.
13500 paired-reads (in 3262 unique pairings) successfully merged out of 17998 (in 6643 pairings) input.
18807 paired-reads (in 3926 unique pairings) successfully merged out of 23103 (in 7020 pairings) input.
16419 paired-reads (in 3690 unique pairings) successfully merged out of 20214 (in 6399 pairings) input.
18123 paired-reads (in 3866 unique pairings) successfully merged out of 23419 (in 7730 pairings) input.
15916 paired-reads (in 3539 unique pairings) successfully merged out of 19201 (in 5950 pairings) input.
19932 paired-reads (in 4524 unique pairings) successfully merged out of 25372 (in 8210 pairings) input.
20628 paired-reads (in 4461 unique pairings) successfully merged out of 25097 (in 7513 pairings) input.
41 paired-reads (in 21 unique pairings) successfully merged out of 50 (in 28 pairings) input.
980 paired-reads (in 433 unique pairings) successfully merged out of 1179 (in 576 pairings) input.
8246 paired-reads (in 2222 unique pairings) successfully merged out of 10683 (in 4079 pairings) input.
11806 paired-reads (in 2813 unique pairings) successfully merged out of 14271 (in 4634 pairings) input.
9576 paired-reads (in 2585 unique pairings) successfully merged out of 10976 (in 3484 pairings) input.
5400 paired-reads (in 1734 unique pairings) successfully merged out of 6355 (in 2386 pairings) input.
16238 paired-reads (in 3756 unique pairings) successfully merged out of 19792 (in 6263 pairings) input.
23900 paired-reads (in 4545 unique pairings) successfully merged out of 30154 (in 9036 pairings) input.
326 paired-reads (in 165 unique pairings) successfully merged out of 388 (in 212 pairings) input.
14036 paired-reads (in 3248 unique pairings) successfully merged out of 17782 (in 6003 pairings) input.
6148 paired-reads (in 1932 unique pairings) successfully merged out of 8535 (in 3718 pairings) input.
11043 paired-reads (in 3034 unique pairings) successfully merged out of 14514 (in 5549 pairings) input.
18478 paired-reads (in 4040 unique pairings) successfully merged out of 21975 (in 6254 pairings) input.
11202 paired-reads (in 2821 unique pairings) successfully merged out of 14062 (in 4987 pairings) input.
39050 paired-reads (in 6140 unique pairings) successfully merged out of 47702 (in 12278 pairings) input.
35607 paired-reads (in 5684 unique pairings) successfully merged out of 44221 (in 12118 pairings) input.
12606 paired-reads (in 3175 unique pairings) successfully merged out of 15023 (in 4853 pairings) input.
13664 paired-reads (in 3331 unique pairings) successfully merged out of 16082 (in 4943 pairings) input.
7798 paired-reads (in 2254 unique pairings) successfully merged out of 10101 (in 3913 pairings) input.
12505 paired-reads (in 3114 unique pairings) successfully merged out of 16001 (in 5708 pairings) input.
10929 paired-reads (in 2736 unique pairings) successfully merged out of 12809 (in 4046 pairings) input.
7038 paired-reads (in 2048 unique pairings) successfully merged out of 8827 (in 3395 pairings) input.
5968 paired-reads (in 1898 unique pairings) successfully merged out of 7122 (in 2627 pairings) input.
3649 paired-reads (in 1258 unique pairings) successfully merged out of 4371 (in 1721 pairings) input.
9934 paired-reads (in 2723 unique pairings) successfully merged out of 11898 (in 3973 pairings) input.
24574 paired-reads (in 4518 unique pairings) successfully merged out of 29822 (in 8252 pairings) input.
15226 paired-reads (in 3351 unique pairings) successfully merged out of 18199 (in 5486 pairings) input.
9442 paired-reads (in 2511 unique pairings) successfully merged out of 11930 (in 4314 pairings) input.
6089 paired-reads (in 1835 unique pairings) successfully merged out of 7757 (in 3044 pairings) input.
9005 paired-reads (in 2561 unique pairings) successfully merged out of 11256 (in 4195 pairings) input.
20 paired-reads (in 10 unique pairings) successfully merged out of 35 (in 20 pairings) input.
20693 paired-reads (in 4240 unique pairings) successfully merged out of 25774 (in 8106 pairings) input.
3450 paired-reads (in 1146 unique pairings) successfully merged out of 4227 (in 1713 pairings) input.
18966 paired-reads (in 4271 unique pairings) successfully merged out of 23666 (in 7583 pairings) input.
17977 paired-reads (in 4104 unique pairings) successfully merged out of 22055 (in 6847 pairings) input.
16384 paired-reads (in 3814 unique pairings) successfully merged out of 19577 (in 5863 pairings) input.
9508 paired-reads (in 2664 unique pairings) successfully merged out of 11777 (in 4186 pairings) input.
3912 paired-reads (in 1254 unique pairings) successfully merged out of 4754 (in 1880 pairings) input.
15882 paired-reads (in 3450 unique pairings) successfully merged out of 19018 (in 5694 pairings) input.
18738 paired-reads (in 3747 unique pairings) successfully merged out of 23048 (in 6960 pairings) input.
9984 paired-reads (in 2838 unique pairings) successfully merged out of 13209 (in 5238 pairings) input.
10528 paired-reads (in 2660 unique pairings) successfully merged out of 12196 (in 3745 pairings) input.
11177 paired-reads (in 2866 unique pairings) successfully merged out of 14133 (in 5022 pairings) input.
13810 paired-reads (in 3317 unique pairings) successfully merged out of 17047 (in 5678 pairings) input.
8231 paired-reads (in 2258 unique pairings) successfully merged out of 9958 (in 3463 pairings) input.
5025 paired-reads (in 1645 unique pairings) successfully merged out of 5966 (in 2326 pairings) input.
9221 paired-reads (in 2648 unique pairings) successfully merged out of 11194 (in 3988 pairings) input.
633 paired-reads (in 320 unique pairings) successfully merged out of 804 (in 442 pairings) input.
7238 paired-reads (in 2190 unique pairings) successfully merged out of 8625 (in 3097 pairings) input.
62250 paired-reads (in 6864 unique pairings) successfully merged out of 70751 (in 12411 pairings) input.
14110 paired-reads (in 3102 unique pairings) successfully merged out of 16650 (in 4917 pairings) input.
52883 paired-reads (in 7000 unique pairings) successfully merged out of 65410 (in 15886 pairings) input.
33 paired-reads (in 16 unique pairings) successfully merged out of 38 (in 20 pairings) input.
10508 paired-reads (in 2502 unique pairings) successfully merged out of 11881 (in 3392 pairings) input.
30748 paired-reads (in 5331 unique pairings) successfully merged out of 37966 (in 10418 pairings) input.
70838 paired-reads (in 9330 unique pairings) successfully merged out of 88602 (in 22360 pairings) input.
100162 paired-reads (in 10826 unique pairings) successfully merged out of 121280 (in 25574 pairings) input.
101525 paired-reads (in 10665 unique pairings) successfully merged out of 121942 (in 24091 pairings) input.
88165 paired-reads (in 9565 unique pairings) successfully merged out of 105062 (in 19839 pairings) input.
68401 paired-reads (in 8479 unique pairings) successfully merged out of 79299 (in 14812 pairings) input.
23420 paired-reads (in 4723 unique pairings) successfully merged out of 27922 (in 7659 pairings) input.
100803 paired-reads (in 8594 unique pairings) successfully merged out of 112666 (in 15149 pairings) input.
81733 paired-reads (in 8079 unique pairings) successfully merged out of 89185 (in 11677 pairings) input.
116339 paired-reads (in 11219 unique pairings) successfully merged out of 144035 (in 30796 pairings) input.
17835 paired-reads (in 3816 unique pairings) successfully merged out of 20302 (in 5293 pairings) input.
20040 paired-reads (in 4190 unique pairings) successfully merged out of 23017 (in 6100 pairings) input.
27646 paired-reads (in 5095 unique pairings) successfully merged out of 33305 (in 8916 pairings) input.
173621 paired-reads (in 13222 unique pairings) successfully merged out of 205960 (in 33886 pairings) input.
12668 paired-reads (in 2764 unique pairings) successfully merged out of 14368 (in 3759 pairings) input.
136383 paired-reads (in 12239 unique pairings) successfully merged out of 168561 (in 34492 pairings) input.
90094 paired-reads (in 10212 unique pairings) successfully merged out of 115849 (in 28070 pairings) input.
69894 paired-reads (in 9316 unique pairings) successfully merged out of 97909 (in 29196 pairings) input.
29085 paired-reads (in 5226 unique pairings) successfully merged out of 34226 (in 8317 pairings) input.
56302 paired-reads (in 8184 unique pairings) successfully merged out of 79037 (in 25173 pairings) input.
52800 paired-reads (in 6798 unique pairings) successfully merged out of 61474 (in 12361 pairings) input.
56089 paired-reads (in 8195 unique pairings) successfully merged out of 77020 (in 23314 pairings) input.
24663 paired-reads (in 4629 unique pairings) successfully merged out of 29089 (in 7524 pairings) input.
8383 paired-reads (in 1693 unique pairings) successfully merged out of 9277 (in 2161 pairings) input.
56816 paired-reads (in 7396 unique pairings) successfully merged out of 70223 (in 16860 pairings) input.
42455 paired-reads (in 7088 unique pairings) successfully merged out of 59443 (in 20005 pairings) input.
15400 paired-reads (in 3768 unique pairings) successfully merged out of 22457 (in 9154 pairings) input.
19362 paired-reads (in 4335 unique pairings) successfully merged out of 29533 (in 12379 pairings) input.
34018 paired-reads (in 6440 unique pairings) successfully merged out of 51072 (in 19166 pairings) input.
50559 paired-reads (in 7921 unique pairings) successfully merged out of 72350 (in 23630 pairings) input.
11734 paired-reads (in 3135 unique pairings) successfully merged out of 15494 (in 5904 pairings) input.
45590 paired-reads (in 7472 unique pairings) successfully merged out of 66102 (in 23210 pairings) input.
59905 paired-reads (in 7721 unique pairings) successfully merged out of 77710 (in 20790 pairings) input.
48170 paired-reads (in 7471 unique pairings) successfully merged out of 67751 (in 22190 pairings) input.
25976 paired-reads (in 5332 unique pairings) successfully merged out of 39831 (in 15903 pairings) input.
15543 paired-reads (in 3867 unique pairings) successfully merged out of 28221 (in 13992 pairings) input.
27461 paired-reads (in 5334 unique pairings) successfully merged out of 41710 (in 16124 pairings) input.
31184 paired-reads (in 5592 unique pairings) successfully merged out of 43565 (in 14556 pairings) input.
26069 paired-reads (in 5427 unique pairings) successfully merged out of 38217 (in 14831 pairings) input.
38020 paired-reads (in 5358 unique pairings) successfully merged out of 51044 (in 14709 pairings) input.
17347 paired-reads (in 3484 unique pairings) successfully merged out of 19501 (in 4715 pairings) input.
10651 paired-reads (in 2756 unique pairings) successfully merged out of 12524 (in 3949 pairings) input.
23807 paired-reads (in 4868 unique pairings) successfully merged out of 29994 (in 9061 pairings) input.
</code></pre>
</details>

<br/>

```{r}
head(mergers[[2]])
```

<br/>

```{r, echo=FALSE, eval=TRUE}
merge_y0 <- read.table("files/trepo/dada2/tables/ssu_merge.txt",
                       header = TRUE, row.names = 1)
datatable(merge_y0, width = "80%",
          caption = htmltools::tags$caption(
            style = 'caption-side: bottom; text-align: left;',
            'Table: ', htmltools::em('Number of denoised reads that were merged.')),
          extensions = 'Buttons', options = list(
            scrollX = TRUE,
            dom = 'Blfrtip',
            buttons = c('copy', 'csv', 'excel'),
            pageLength = 5,
            lengthMenu = list(c(5, 10, -1), c("5", "10", "All"))
            )
          )
```


```{r}
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
```

```{r, echo=FALSE, eval=TRUE}
seqtab <- c(312, 119067)
seqtab
```

```{r}
table(nchar(getSequences(seqtab)))
```

```
  251   254   257   265   271   275   276   280   284   285   290   297   310 
    1     1     1    13     3     1     1     1    10     2     1     1     1 
  314   315   322   323   324   326   327   330   331   335   342   344   345 
    1     2     4     1     1     2     1     2     1     1     3     1     2 
  346   347   350   351   353   357   361   362   364   365   366   367   368 
    1     1     2     1     1     3     1     7     9    14    64    83   803 
  369   370   371   372   373   374   375   376   377   378   379   380 
  405  4188  3664  6256  7675 62269  7456  2616 14868  7233   927   461 
```

```{r, echo=FALSE}
read_length <-  data.frame(nchar(getSequences(seqtab)))
colnames(read_length) <- "length"
plot_a <- qplot(length, data = read_length, 
                        geom = "histogram", binwidth = 1,
                        xlab = "read length", ylab = "total variants", 
                        xlim = c(360,390)) 
ggsave("figures/ssu_read_length_before_pseudo_pooled.png", 
         plot_a, width = 7, height = 3)
```

```{r, echo=FALSE, layout="l-body-outset", eval=TRUE}
system("cp files/trepo/dada2/figures/ssu_read_length_before_pseudo_pooled.png include/trepo/dada2/ssu_read_length_before_pseudo_pooled.png")

knitr::include_graphics("include/trepo/dada2/ssu_read_length_before_pseudo_pooled.png")
```
<small><em>Distribution of read lengths by number of samples.</em></small>

The sequence table is a matrix with rows corresponding to (and named by) the samples, and columns corresponding to (and named by) the sequence variants. We have `r (seqtab)[2]` sequence variants but also a range of sequence lengths. Since many of these sequence variants are singletons or doubletons, we will just select a range that corresponds to the expected amplicon length and eliminate the spurious reads. 

<aside>
The values you select should be based on the sequence table generated above.
</aside>

```{r}
seqtab.2 <- seqtab[,nchar(colnames(seqtab)) %in% seq(370,380)]
dim(seqtab.2)

saveRDS(seqtab.2, "rdata/seqtab_pseudo_pooled.rds")
```


```{r, echo=FALSE, eval=TRUE}
seqtab2 <-  c(312, 117613)
seqtab2
```

```{r}
table(nchar(getSequences(seqtab.2)))
```

```
  370   371   372   373   374   375   376   377   378   379   380 
 4188  3664  6256  7675 62269  7456  2616 14868  7233   927   461 
```

After removing the extreme length variants, we have `r (seqtab2)[2]`, a reduction of `r (seqtab)[2] - (seqtab2)[2]` sequence variants.

```{r, echo=FALSE}
saveRDS(seqtab.2, "ssu_seqtab.rds")
```

# 5. Remove Chimeras

Even though the `dada` method corrects substitution and indel errors,  chimeric sequences remain. According to the DADA2 documentation, the accuracy of sequence variants after denoising makes identifying chimeric ASVs simpler than when dealing with fuzzy OTUs. Chimeric sequences are identified if they can be *exactly reconstructed* by combining a left-segment and a right-segment from two more abundant *parent* sequences.

```{r}
seqtab.nochim <- removeBimeraDenovo(seqtab.2,  method = "pooled", 
                                    multithread = 20, verbose = TRUE)
dim(seqtab.nochim)
```

```
Identified 69692 bimeras out of 117613 input sequences.
```

```{r, echo=FALSE, eval=TRUE}
seqtab3 <- c(312, 47921)
seqtab3
```

```{r}
sum(seqtab.nochim)/sum(seqtab.2)
```

```{r, echo=FALSE, eval=TRUE}
chim_rem <- 0.8818896
0.8818896
```

```
  370   371   372   373   374   375   376   377   378   379   380 
 2488  1616  2792  3728 22922  3529  1399  4150  4543   577   177 
```

Chimera checking removed an additional `r (seqtab2)[2] - (seqtab3)[2]` sequence variants however, when we account for the abundances of each variant, we see chimeras accounts for about `r 100*(1-chim_rem)`% of the merged sequence reads. Not bad.

The last thing we want to do is write the sequence table to an `RDS` file.

```{r}
saveRDS(seqtab.nochim, "ssu_seqtab.nochim.rds")
```

Now, let's look at the distribution of ASVs by the number of samples

```{r}
table(colSums(seqtab.nochim > 0))
```

```
    1     2     3     4     5     6     7     8     9    10    11    12    13 
27657  1261   654   577   520   500   465   415   417   455   406   347   391 
   14    15    16    17    18    19    20    21    22    23    24    25    26 
  359   326   309   326   299   291   278   250   299   266   252   226   243 
   27    28    29    30    31    32    33    34    35    36    37    38    39 
  202   237   235   230   229   228   188   221   177   196   160   156   172 
   40    41    42    43    44    45    46    47    48    49    50    51    52 
  167   159   125   155   133   147   119   121   127   137   114   132   121 
   53    54    55    56    57    58    59    60    61    62    63    64    65 
  110   124   111    94   108    92    79   109    84   105    79    93    97 
   66    67    68    69    70    71    72    73    74    75    76    77    78 
   93    87    82    91    97    65    82    77    63    71    74    62    57 
   79    80    81    82    83    84    85    86    87    88    89    90    91 
   69    66    62    51    73    51    59    54    63    46    50    55    61 
   92    93    94    95    96    97    98    99   100   101   102   103   104 
   44    51    50    34    48    43    54    34    54    45    41    40    43 
  105   106   107   108   109   110   111   112   113   114   115   116   117 
   41    42    37    32    44    42    42    35    31    34    40    25    37 
  118   119   120   121   122   123   124   125   126   127   128   129   130 
   38    27    20    21    35    38    37    24    26    22    19    24    17 
  131   132   133   134   135   136   137   138   139   140   141   142   143 
   14    26    24    17    20    32    16    20    27    18    28    21    13 
  144   145   146   147   148   149   150   151   152   153   154   155   156 
   25    14    18    19    16    13    18    11    14    13    11    16    17 
  157   158   159   160   161   162   163   164   165   166   167   168   169 
   18    19    17    10     9    14    10    19    16     6    11    12    15 
  170   171   172   173   174   175   176   177   178   179   180   181   182 
   15     6    12    15     8     7    10    11     5    10     4    12     8 
  183   184   185   186   187   188   189   190   191   192   193   194   195 
   11    13     6     9     4     6     8     5     4     7     6     8     5 
  196   197   198   199   200   201   202   203   204   205   206   207   208 
    4     5     4     7     8     5     8     5     4    11     6     4     4 
  209   210   211   212   213   214   215   216   217   218   219   220   221 
    8     4     3     6     5     5     6     5     4     4     5     5     1 
  222   223   224   225   226   227   228   229   230   231   232   233   234 
    8     4     4     6     4     7     6     5     9     6     3     3     8 
  235   236   237   238   239   240   241   242   243   244   245   246   247 
    2     3     2     7     2     6     3     2     4     1     2     1     3 
  248   249   250   251   252   253   254   255   256   257   258   259   260 
    2     1     2     2     3     7     1     2     2     1     3     4     3 
  261   262   263   264   265   266   267   268   269   270   271   272   273 
    7     4     4     8     3     1     1     5     1     4     6     2     5 
  274   275   276   277   278   279   280   281   283   284   285   286   287 
    2     2     6     4     1     5     1     1     4     3     3     3     3 
  288   290   291   292   293   294   295   296   297   298   300   301   302 
    2     4     2     2     2     6     3     2     1     8     2     3     2 
  303   304   305   306   307 
    3     1     1     1     1 
```

```{r, echo=FALSE, eval=TRUE}
sample_by_asvs <- read.table("include/trepo/dada2/ssu_seqtab_sum_samples_by_asvs.txt", 
                            header = TRUE, sep = "\t")
sample_by_asvs$Freq <- log(sample_by_asvs$Freq)
plot(sample_by_asvs, type = "s", main = "Total ASVs per sample", 
     xlab = "Number of Samples", ylab = "Number of ASVs")
```

And the distribution of total ASVs across samples. 

```{r}
table(rowSums(seqtab.nochim > 0))
```


```
   2    4    8   16   18  136  254  350  851  882  957  975 1043 1140 1150 1246 
   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 
1307 1345 1356 1369 1410 1416 1429 1462 1471 1518 1524 1536 1538 1540 1542 1546 
   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 
1566 1600 1623 1637 1638 1652 1674 1685 1689 1699 1707 1725 1742 1748 1759 1761 
   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 
1776 1802 1810 1812 1813 1815 1819 1846 1847 1853 1867 1870 1872 1882 1884 1887 
   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 
1897 1909 1913 1920 1927 1942 1945 1947 1948 1960 1963 1975 1979 1985 1992 1993 
   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 
2000 2015 2019 2023 2025 2031 2033 2041 2049 2054 2060 2064 2077 2103 2108 2110 
   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 
2116 2117 2121 2138 2144 2146 2159 2160 2161 2165 2169 2173 2177 2193 2195 2207 
   1    1    1    1    1    1    2    1    1    1    1    1    1    1    1    1 
2221 2222 2241 2254 2257 2263 2265 2266 2307 2318 2322 2340 2343 2348 2352 2355 
   1    1    1    1    2    1    1    1    1    1    1    1    1    1    1    1 
2365 2376 2380 2393 2416 2434 2444 2446 2457 2459 2487 2493 2497 2512 2525 2557 
   2    2    1    1    2    1    1    1    1    2    1    1    1    1    1    1 
2570 2577 2580 2583 2594 2600 2602 2641 2664 2693 2694 2699 2703 2711 2715 2736 
   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 
2750 2754 2757 2758 2784 2791 2796 2803 2821 2822 2854 2862 2867 2868 2889 2896 
   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 
2950 2958 2974 3001 3013 3028 3039 3047 3060 3073 3078 3082 3091 3092 3093 3107 
   1    1    1    2    1    1    1    1    1    1    1    1    1    1    1    1 
3108 3112 3116 3141 3157 3170 3177 3191 3200 3202 3204 3230 3245 3251 3267 3270 
   1    1    2    1    1    1    1    1    1    1    2    1    1    1    1    1 
3280 3319 3348 3351 3390 3400 3411 3418 3422 3437 3452 3461 3471 3472 3499 3501 
   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 
3523 3543 3568 3569 3573 3575 3592 3601 3619 3621 3623 3627 3630 3631 3642 3647 
   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 
3666 3689 3704 3723 3727 3730 3734 3785 3792 3815 3836 3842 3861 3883 3977 3987 
   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 
4022 4028 4033 4035 4143 4247 4251 4297 4352 4372 4665 4675 4691 4699 4723 4724 
   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 
4751 4891 4970 4990 5018 5058 5083 5216 5229 5238 5320 5330 5670 5763 6041 6112 
   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 
6147 6150 6169 6284 6706 6903 6951 7013 7247 7274 7353 8001 8033 8186 8756 
   1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 
```

```{r, echo=FALSE, eval=TRUE}
asv_by_sample <- read.table("include/trepo/dada2/ssu_seqtab_sum_asvs_by_samples.txt", 
                            header = TRUE, sep = "\t")
hist(asv_by_sample$Total_ASVs, main = "Total ASVs per sample", 
     xlab = "Number of ASVs", ylab = "Number of Samples", 
     border = "#323232", col = "#56B4E9", 
     xlim = c (0,10000), ylim = c(0,100), breaks = 10)
```

# 6. Assign Taxonomy

The `assignTaxonomy` command implements the naive Bayesian classifier, so for reproducible results you need to set a random number seed (see issue [#538](https://github.com/benjjneb/dada2/issues/538)). We did this at the beginning of the workflow. For taxonomic assignment, we are using the Silva version 138 [@quast2012silva]. The developers of DADA2 maintain a [formatted version of the database](https://benjjneb.github.io/dada2/training.html).

We will read in the `RDS` file containing the sequence table saved above. We also need to run `remove(list = ls())` command, otherwise the final image we save will be huge. This way, the image only contains the sample data, seqtab, and taxtab *after* running `removeBimeraDenovo`.

```{r}
remove(list = ls())
seqtab <- readRDS("rdata/seqtab.nochim.rds")
```

And then native Bayesian classifier against the Silva database.

```{r}
tax_silva <- assignTaxonomy(seqtab, "TAXONOMY_FILES/silva_nr_v138_train_set.fa.gz", 
                            multithread = TRUE, verbose = TRUE)
saveRDS(tax_silva, "rdata/tax_silva_pseudo_pooled.rds")
```


```{r, echo=FALSE}
# Classify with IdTAX db
remove(list = ls())
seqtab <- readRDS("ssu_seqtab.nochim.rds")

tax_silva <- assignTaxonomy(seqtab, "silva_nr_v138_train_set.fa.gz", multithread = TRUE)

dna <- DNAStringSet(getSequences(seqtab)) # Create a DNAStringSet from the ASVs
load("SILVA_SSU_r138_2018.RData")
ids <- IdTaxa(dna, trainingSet, strand = "top", processors = NULL, verbose = FALSE) # use all processors
ranks <- c("domain", "phylum", "class", "order", "family", "genus") # ranks of interest
# Convert the output object of class "Taxa" to a matrix analogous to the output from assignTaxonomy
tax_id <- t(sapply(ids, function(x) {
        m <- match(ranks, x$rank)
        taxa <- x$taxon[m]
        taxa[startsWith(taxa, "unclassified_")] <- NA
        taxa
}))
colnames(tax_id) <- ranks; rownames(tax_id) <- getSequences(seqtab)

save.image("ssu_data_prep_wf_ALT.rdata")
saveRDS(seqtab, "ssu_seqtab.nochim.2.rds")
saveRDS(tax_silva, "ssu_tax_silva.rds")
saveRDS(tax_id, "ssu_tax_id.rds")
```

# 7. Summary

```{r, echo=FALSE}
cut_all <- read.table("files/trepo/dada2/tables/ssu_cut.txt", header = TRUE)
cut_all$SampleId  <- gsub(x = cut_all$SampleId, pattern = "_R1", replacement = "") 

filt_all <- read.table("files/trepo/dada2/tables/ssu_filt.txt", header = TRUE)
filt_all$reads.in <- NULL
filt_all$SampleId  <- gsub(x = filt_all$SampleId, pattern = "_R1", replacement = "") 
filt_all <- filt_all %>% dplyr::rename("filtered" = 2)

denoise_all <- read.table("files/trepo/dada2/tables/ssu_denoise.txt", header = TRUE)

merge_all <- read.table("files/trepo/dada2/tables/ssu_merge.txt", header = TRUE)

merge_all[,c(2:3)] <- NULL

seqtab.nochim <- readRDS("files/trepo/dada2/rdata/seqtab.nochim_pseudo_pooled.rds")
nochime <- data.frame(rowSums(seqtab.nochim))
nochime <- nochime %>% dplyr::rename("nochim" = 1)
nochime <- nochime %>% tibble::rownames_to_column("SampleId")

track_changes <- dplyr::left_join(cut_all, filt_all) %>%
                 dplyr::left_join(., denoise_all) %>%
                 dplyr::left_join(., merge_all) %>%
                 dplyr::left_join(., nochime)

track_changes$perc_remain <- track_changes$nochim/track_changes$raw
track_changes$perc_remain <- round(track_changes$perc_remain, digits = 3)
write.table(track_changes, "files/trepo/dada2/tables/ssu_read_changes.txt", 
                            sep = "\t", quote = FALSE, row.names = FALSE)
```

<br/>

```{r, echo=FALSE, eval=TRUE, layout="l-body-outset"}
track_changes <- read.table("files/trepo/dada2/tables/ssu_read_changes.txt",
                       header = TRUE, row.names = 1)
track_changes <- track_changes %>% tibble::rownames_to_column("SampleID")
datatable(track_changes, width = "80%", rownames = FALSE,
          caption = htmltools::tags$caption(
            style = 'caption-side: bottom; text-align: left;',
            'Table: ', htmltools::em('Changes from raw to final read counts.')),
          extensions = 'Buttons', options = list(
            scrollX = TRUE,
            dom = 'Blfrtip',
            buttons = c('copy', 'csv', 'excel'),
            pageLength = 5,
            lengthMenu = list(c(5, 10, -1), c("5", "10", "All"))
            )
          )
```

# R Session Information & Code

This workflow was run on the [Smithsonian High Performance Cluster (SI/HPC)](https://doi.org/10.25572/SIHPC), Smithsonian Institution. Below are the specific packages and versions used in this workflow using both `sessionInfo()` and `devtools::session_info()`. Click the arrow to see the details.

<details markdown="1">
<summary>Show/hide  R Session Info</summary>
```
R version 4.0.0 (2020-04-24)
Platform: x86_64-conda_cos6-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /home/scottjj/miniconda3/envs/R-4/lib/libopenblasp-r0.3.9.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] DECIPHER_2.16.1             RSQLite_2.2.0              
 [3] gridExtra_2.3               ggplot2_3.3.2              
 [5] ShortRead_1.46.0            GenomicAlignments_1.24.0   
 [7] SummarizedExperiment_1.18.2 DelayedArray_0.14.1        
 [9] matrixStats_0.56.0          Biobase_2.48.0             
[11] Rsamtools_2.4.0             GenomicRanges_1.40.0       
[13] GenomeInfoDb_1.24.2         Biostrings_2.56.0          
[15] XVector_0.28.0              IRanges_2.22.2             
[17] S4Vectors_0.26.1            BiocParallel_1.22.0        
[19] BiocGenerics_0.34.0         dada2_1.16.0               
[21] Rcpp_1.0.5                 

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.0       purrr_0.3.4            reshape2_1.4.4        
 [4] lattice_0.20-41        colorspace_1.4-1       vctrs_0.3.4           
 [7] generics_0.0.2         blob_1.2.1             rlang_0.4.7           
[10] pillar_1.4.6           glue_1.4.2             withr_2.2.0           
[13] DBI_1.1.0              bit64_4.0.2            RColorBrewer_1.1-2    
[16] jpeg_0.1-8.1           GenomeInfoDbData_1.2.3 lifecycle_0.2.0       
[19] plyr_1.8.6             stringr_1.4.0          zlibbioc_1.34.0       
[22] munsell_0.5.0          gtable_0.3.0           hwriter_1.3.2         
[25] memoise_1.1.0          latticeExtra_0.6-29    scales_1.1.1          
[28] RcppParallel_5.0.2     bit_4.0.4              digest_0.6.25         
[31] png_0.1-7              stringi_1.5.3          dplyr_1.0.2           
[34] tools_4.0.0            bitops_1.0-6           magrittr_1.5          
[37] RCurl_1.98-1.2         tibble_3.0.3           crayon_1.3.4          
[40] pkgconfig_2.0.3        ellipsis_0.3.1         Matrix_1.2-18         
[43] R6_2.4.1               compiler_4.0.0        
â”€ Session info â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 setting  value                       
 version  R version 4.0.0 (2020-04-24)
 os       CentOS Linux 7 (Core)       
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       America/New_York            
 date     2020-12-04                  

â”€ Packages â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 package              * version  date       lib source        
 assertthat             0.2.1    2019-03-21 [1] CRAN (R 4.0.0)
 backports              1.1.9    2020-08-24 [1] CRAN (R 4.0.0)
 Biobase              * 2.48.0   2020-04-27 [1] Bioconductor  
 BiocGenerics         * 0.34.0   2020-04-27 [1] Bioconductor  
 BiocParallel         * 1.22.0   2020-04-27 [1] Bioconductor  
 Biostrings           * 2.56.0   2020-04-27 [1] Bioconductor  
 bit                    4.0.4    2020-08-04 [1] CRAN (R 4.0.0)
 bit64                  4.0.2    2020-07-30 [1] CRAN (R 4.0.0)
 bitops                 1.0-6    2013-08-17 [1] CRAN (R 4.0.0)
 blob                   1.2.1    2020-01-20 [1] CRAN (R 4.0.0)
 callr                  3.4.4    2020-09-07 [1] CRAN (R 4.0.0)
 cli                    2.0.2    2020-02-28 [1] CRAN (R 4.0.0)
 colorspace             1.4-1    2019-03-18 [1] CRAN (R 4.0.0)
 crayon                 1.3.4    2017-09-16 [1] CRAN (R 4.0.0)
 dada2                * 1.16.0   2020-04-27 [1] Bioconductor  
 DBI                    1.1.0    2019-12-15 [1] CRAN (R 4.0.0)
 DECIPHER             * 2.16.1   2020-05-11 [1] Bioconductor  
 DelayedArray         * 0.14.1   2020-07-14 [1] Bioconductor  
 desc                   1.2.0    2018-05-01 [1] CRAN (R 4.0.0)
 devtools               2.3.1    2020-07-21 [1] CRAN (R 4.0.0)
 digest                 0.6.25   2020-02-23 [1] CRAN (R 4.0.0)
 dplyr                  1.0.2    2020-08-18 [1] CRAN (R 4.0.0)
 ellipsis               0.3.1    2020-05-15 [1] CRAN (R 4.0.0)
 fansi                  0.4.1    2020-01-08 [1] CRAN (R 4.0.0)
 fs                     1.5.0    2020-07-31 [1] CRAN (R 4.0.0)
 generics               0.0.2    2018-11-29 [1] CRAN (R 4.0.0)
 GenomeInfoDb         * 1.24.2   2020-06-15 [1] Bioconductor  
 GenomeInfoDbData       1.2.3    2020-05-22 [1] Bioconductor  
 GenomicAlignments    * 1.24.0   2020-04-27 [1] Bioconductor  
 GenomicRanges        * 1.40.0   2020-04-27 [1] Bioconductor  
 ggplot2              * 3.3.2    2020-06-19 [1] CRAN (R 4.0.0)
 glue                   1.4.2    2020-08-27 [1] CRAN (R 4.0.0)
 gridExtra            * 2.3      2017-09-09 [1] CRAN (R 4.0.0)
 gtable                 0.3.0    2019-03-25 [1] CRAN (R 4.0.0)
 hwriter                1.3.2    2014-09-10 [1] CRAN (R 4.0.0)
 IRanges              * 2.22.2   2020-05-21 [1] Bioconductor  
 jpeg                   0.1-8.1  2019-10-24 [1] CRAN (R 4.0.0)
 lattice                0.20-41  2020-04-02 [1] CRAN (R 4.0.0)
 latticeExtra           0.6-29   2019-12-19 [1] CRAN (R 4.0.0)
 lifecycle              0.2.0    2020-03-06 [1] CRAN (R 4.0.0)
 magrittr               1.5      2014-11-22 [1] CRAN (R 4.0.0)
 Matrix                 1.2-18   2019-11-27 [1] CRAN (R 4.0.0)
 matrixStats          * 0.56.0   2020-03-13 [1] CRAN (R 4.0.0)
 memoise                1.1.0    2017-04-21 [1] CRAN (R 4.0.0)
 munsell                0.5.0    2018-06-12 [1] CRAN (R 4.0.0)
 pillar                 1.4.6    2020-07-10 [1] CRAN (R 4.0.0)
 pkgbuild               1.1.0    2020-07-13 [1] CRAN (R 4.0.0)
 pkgconfig              2.0.3    2019-09-22 [1] CRAN (R 4.0.0)
 pkgload                1.1.0    2020-05-29 [1] CRAN (R 4.0.0)
 plyr                   1.8.6    2020-03-03 [1] CRAN (R 4.0.0)
 png                    0.1-7    2013-12-03 [1] CRAN (R 4.0.0)
 prettyunits            1.1.1    2020-01-24 [1] CRAN (R 4.0.0)
 processx               3.4.4    2020-09-03 [1] CRAN (R 4.0.0)
 ps                     1.3.4    2020-08-11 [1] CRAN (R 4.0.0)
 purrr                  0.3.4    2020-04-17 [1] CRAN (R 4.0.0)
 R6                     2.4.1    2019-11-12 [1] CRAN (R 4.0.0)
 RColorBrewer           1.1-2    2014-12-07 [1] CRAN (R 4.0.0)
 Rcpp                 * 1.0.5    2020-07-06 [1] CRAN (R 4.0.0)
 RcppParallel           5.0.2    2020-06-24 [1] CRAN (R 4.0.0)
 RCurl                  1.98-1.2 2020-04-18 [1] CRAN (R 4.0.0)
 remotes                2.2.0    2020-07-21 [1] CRAN (R 4.0.0)
 reshape2               1.4.4    2020-04-09 [1] CRAN (R 4.0.0)
 rlang                  0.4.7    2020-07-09 [1] CRAN (R 4.0.0)
 rprojroot              1.3-2    2018-01-03 [1] CRAN (R 4.0.0)
 Rsamtools            * 2.4.0    2020-04-27 [1] Bioconductor  
 RSQLite              * 2.2.0    2020-01-07 [1] CRAN (R 4.0.0)
 S4Vectors            * 0.26.1   2020-05-16 [1] Bioconductor  
 scales                 1.1.1    2020-05-11 [1] CRAN (R 4.0.0)
 sessioninfo            1.1.1    2018-11-05 [1] CRAN (R 4.0.0)
 ShortRead            * 1.46.0   2020-04-27 [1] Bioconductor  
 stringi                1.5.3    2020-09-09 [1] CRAN (R 4.0.0)
 stringr                1.4.0    2019-02-10 [1] CRAN (R 4.0.0)
 SummarizedExperiment * 1.18.2   2020-07-09 [1] Bioconductor  
 testthat               2.3.2    2020-03-02 [1] CRAN (R 4.0.0)
 tibble                 3.0.3    2020-07-10 [1] CRAN (R 4.0.0)
 tidyselect             1.1.0    2020-05-11 [1] CRAN (R 4.0.0)
 usethis                1.6.1    2020-04-29 [1] CRAN (R 4.0.0)
 vctrs                  0.3.4    2020-08-29 [1] CRAN (R 4.0.0)
 withr                  2.2.0    2020-04-20 [1] CRAN (R 4.0.0)
 XVector              * 0.28.0   2020-04-27 [1] Bioconductor  
 zlibbioc               1.34.0   2020-04-27 [1] Bioconductor  
```
</details>

<br/>


```{r, include=FALSE, eval=TRUE}
remove(list = ls())
```


##  Source Code {.appendix}

The source code for this page can be accessed on GitHub by [clicking this link](https://github.com/tropical-repo/web/blob/master/trepo-dada2.Rmd). Please note, that in order to process the data *and*  build the website, we needed to run the workflow and get the results. Then hard code the results and turn off the individual commands. So the raw file for this page is a bit messy---you have been warned.

## Data Availability {.appendix}

Raw fastq files available on figshare at XXXXXXXX. Trimmed fastq files (primers removed) available through the ENA under project accession number [XXXXXXXX](). Output files from this workflow available on figshare  5at [XXXXXXXX.]().
